<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PRONE">
<title>Differential Expression Analysis • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential Expression Analysis">
<meta property="og:description" content="PRONE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/PRONE.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/Normalization.html">Normalization</a>
    <a class="dropdown-item" href="../articles/Imputation.html">Imputation</a>
    <a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a>
    <a class="dropdown-item" href="../articles/Spike_In_Data.html">How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<link href="Differential_Expression_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="Differential_Expression_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="Differential_Expression_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="Differential_Expression_files/datatables-binding-0.33/datatables.js"></script><link href="Differential_Expression_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="Differential_Expression_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="Differential_Expression_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="Differential_Expression_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="Differential_Expression_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Differential Expression Analysis</h1>
                        <h4 data-toc-skip class="author">Arend Lis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Differential_Expression.Rmd" class="external-link"><code>vignettes/Differential_Expression.Rmd</code></a></small>
      <div class="d-none name"><code>Differential_Expression.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="load-data-tmt">Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a>
data. For more information on how to create the SummarizedExperiment
from a proteomics data set, please refer to the <a href="PRONE.Rmd">“Get
Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="normalize-data">Normalize Data<a class="anchor" aria-label="anchor" href="#normalize-data"></a>
</h2>
<p>In order to compare the performance of different normalization
methods on their ability to detect differentially expressed proteins, we
first perform a simple normalization here. For more details about how to
normalize data and evaluate the normalization appraoches quantitatively
and qualitatively using PRONE, please refer to the <a href="Normalization.Rmd">“Normalization”</a> vignette.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"IRS_on_Median"</span>, <span class="st">"IRS_on_LoessF"</span>, <span class="st">"IRS_on_Quantile"</span><span class="op">)</span>, combination_pattern <span class="op">=</span> <span class="st">"_on_"</span><span class="op">)</span></span>
<span><span class="co">#&gt; RobNorm normalization not yet performed. Single RobNorm normalization performed now.</span></span>
<span><span class="co">#&gt; RobNorm completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on RobNorm-normalized data completed.</span></span>
<span><span class="co">#&gt; Median normalization not yet performed. Single Median normalization performed now.</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on Median-normalized data completed.</span></span>
<span><span class="co">#&gt; LoessF normalization not yet performed. Single LoessF normalization performed now.</span></span>
<span><span class="co">#&gt; LoessF completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on LoessF-normalized data completed.</span></span>
<span><span class="co">#&gt; Quantile normalization not yet performed. Single Quantile normalization performed now.</span></span>
<span><span class="co">#&gt; Quantile completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on Quantile-normalized data completed.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h2>
<p>After having performed normalization and evaluated the different
normalization methods via qualitative and quantitative analysis,
differential expression analysis can be used to further analyze the
differences of the normalization methods.</p>
<p>However before, you need to remove the reference samples in case of a
TMT experiment. This can be easily done with the function
<code><a href="../reference/remove_reference_samples.html">remove_reference_samples()</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_reference_samples.html">remove_reference_samples</a></span><span class="op">(</span><span class="va">se_norm</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 reference samples removed from the SummarizedExperiment object.</span></span></code></pre></div>
<div class="section level4">
<h4 id="run-de-analysis">Run DE Analysis<a class="anchor" aria-label="anchor" href="#run-de-analysis"></a>
</h4>
<p>First, you need to specify the comparisons you want to perform in DE
analysis. For this, the function <code>specifc_comparisons()</code> was
developed which helps to build the right comparison strings. For
instance, if you are having a condition column “Condition” in your meta
data that is built upon two individual groups, such as “diabetic_3d”
referring to diabetic samples at day 3 after operation, you can simply
specify sep = “_” and the method only considers comparisons where at
least one of the group remains static.</p>
<p>However, you can also just simply create a vector of comparisons to
ensure the correct order and handle this vector over to the DE analysis
method.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/specify_comparisons.html">specify_comparisons</a></span><span class="op">(</span><span class="va">se_norm</span>, condition <span class="op">=</span> <span class="st">"Group"</span>, sep <span class="op">=</span> <span class="cn">NULL</span>, control <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PTB-HC"</span>, <span class="st">"TBL-HC"</span>, <span class="st">"TBL-PTB"</span>, <span class="st">"Rx-PTB"</span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="../reference/run_DE.html">run_DE()</a></code> performs the DE analysis on the
selected SummarizedExperiment and comparisons. DE analysis can be
performed on multiple assays (normalization methods) at once using the
already known “ain” parameter. The condition of the SummarizedExperiment
object, specified at the beginning, can be used (condition = NULL) or
any other column of the meta data can specified. Three methods are
available for DE analysis: limma <span class="citation">(Ritchie et al.
2015)</span>, DEqMS <span class="citation">(Zhu et al. 2020)</span>, and
ROTS <span class="citation">(Suomi et al. 2017)</span>. The meaning of
the other parameters can be extracted from the documentation of the
method.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DE.html">run_DE</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">se_norm</span>, </span>
<span>                     comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>                     ain <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     DE_method <span class="op">=</span> <span class="st">"limma"</span>, </span>
<span>                     covariate <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     logFC <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     logFC_up <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     logFC_down <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                     p_adj <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     trend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     robust <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 <span class="op">)</span> </span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; DE Analysis will not be performed on raw data.</span></span>
<span><span class="co">#&gt; log2: DE analysis completed.</span></span>
<span><span class="co">#&gt; RobNorm: DE analysis completed.</span></span>
<span><span class="co">#&gt; IRS_on_RobNorm: DE analysis completed.</span></span>
<span><span class="co">#&gt; Median: DE analysis completed.</span></span>
<span><span class="co">#&gt; IRS_on_Median: DE analysis completed.</span></span>
<span><span class="co">#&gt; LoessF: DE analysis completed.</span></span>
<span><span class="co">#&gt; IRS_on_LoessF: DE analysis completed.</span></span>
<span><span class="co">#&gt; Quantile: DE analysis completed.</span></span>
<span><span class="co">#&gt; IRS_on_Quantile: DE analysis completed.</span></span></code></pre></div>
<p>When running ROTS you need to specify the R and K parameters. In
DEqMS, the DEqMS_PSMs_column need to be specified. This can be any
column in the <code>rowData(se)</code>. For more information on the
specific parameters, please refer to the method specific
documentation.</p>
<p>If you want to apply other logFC or p-value threshold, there is no
need to re-run the DE analysis again. With
<code><a href="../reference/apply_thresholds.html">apply_thresholds()</a></code>, you can simply change the threshold
values.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/apply_thresholds.html">apply_thresholds</a></span><span class="op">(</span>de_res <span class="op">=</span> <span class="va">de_res</span>, logFC <span class="op">=</span> <span class="cn">FALSE</span>, p_adj <span class="op">=</span> <span class="cn">TRUE</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualize-de-results">Visualize DE Results<a class="anchor" aria-label="anchor" href="#visualize-de-results"></a>
</h4>
<div class="section level5">
<h5 id="barplot">Barplot<a class="anchor" aria-label="anchor" href="#barplot"></a>
</h5>
<p>To get an overview of the DE results of the different normalization
methods, you can visualize the number of significant DE proteins per
normalization method in a barplot using
<code><a href="../reference/plot_overview_DE_bar.html">plot_overview_DE_bar()</a></code>. This plot can be generated in
different ways by specifying the “plot_type” parameter:</p>
<ul>
<li>single: plot a single barplot for each comparison (default)</li>
<li>facet_comp: facet the barplot by comparison</li>
<li>stacked: stack the number of DE per comparison</li>
<li>facet_regulation: stack the number of DE per comparison but facet by
up- and down-regulated</li>
</ul>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_overview_DE_bar.html">plot_overview_DE_bar</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, plot_type <span class="op">=</span> <span class="st">"facet_regulation"</span><span class="op">)</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"bottom"</span>, legend.direction <span class="op">=</span> <span class="st">"vertical"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>You can also just visualize two specific comparisons:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_overview_DE_bar.html">plot_overview_DE_bar</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>, plot_type <span class="op">=</span> <span class="st">"facet_comp"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="tile-plot">Tile Plot<a class="anchor" aria-label="anchor" href="#tile-plot"></a>
</h5>
<p>You can also get an overview of the DE results in form of a heatmap
using the <code><a href="../reference/plot_overview_DE_tile.html">plot_overview_DE_tile()</a></code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_overview_DE_tile.html">plot_overview_DE_tile</a></span><span class="op">(</span><span class="va">de_res</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="volcano-plots">Volcano Plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h5>
<p>Another option is to generate volcano plots for each comparison. The
function <code><a href="../reference/plot_volcano_DE.html">plot_volcano_DE()</a></code> generates a grid of volcano
plots for each normalization techniques (facet_norm = TRUE) or for each
comparison (facet_comparison = TRUE). A list of volcano plots is
returned.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano_DE.html">plot_volcano_DE</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span>
<span><span class="co">#&gt; $`PTB-HC`</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="heatmap-of-de-results">Heatmap of DE Results<a class="anchor" aria-label="anchor" href="#heatmap-of-de-results"></a>
</h5>
<p>Furthermore, you can visualize the DE results in form of a heatmap.
The function <code><a href="../reference/plot_heatmap_DE.html">plot_heatmap_DE()</a></code> generates a heatmap of the
DE results for a specific comparison and normalization method.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_heatmap_DE.html">plot_heatmap_DE</a></span><span class="op">(</span><span class="va">tuberculosis_TMT_se</span>, <span class="va">tuberculosis_TMT_de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparison <span class="op">=</span> <span class="st">"PTB-HC"</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, pvalue_column <span class="op">=</span> <span class="st">"adj.P.Val"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; &lt;simpleError in stats::hclust(stats::dist(as.matrix(data))): NA/NaN/Inf in foreign function call (arg 10)&gt;</span></span>
<span><span class="co">#&gt; &lt;simpleError in stats::hclust(stats::dist(as.matrix(data))): NA/NaN/Inf in foreign function call (arg 10)&gt;</span></span>
<span><span class="co">#&gt; &lt;simpleError in stats::hclust(stats::dist(as.matrix(data))): NA/NaN/Inf in foreign function call (arg 10)&gt;</span></span>
<span><span class="co">#&gt; &lt;simpleError in stats::hclust(stats::dist(as.matrix(data))): NA/NaN/Inf in foreign function call (arg 10)&gt;</span></span>
<span><span class="co">#&gt; &lt;simpleError in stats::hclust(stats::dist(as.matrix(data))): NA/NaN/Inf in foreign function call (arg 10)&gt;</span></span>
<span><span class="co">#&gt; $log2</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $RobNorm</span></span></code></pre>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $IRS_on_RobNorm</span></span></code></pre>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-12-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Median</span></span></code></pre>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-12-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $IRS_on_Median</span></span></code></pre>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-12-5.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="coverage-of-biomarkers">Coverage of Biomarkers<a class="anchor" aria-label="anchor" href="#coverage-of-biomarkers"></a>
</h5>
<p>In case you know some biomarkers of previous publications that should
be significantly expressed in your biological context, you can check the
number of specified markers that are DE in the different normalization
methods.</p>
<p>TODO: plot_coverage_DE_bar</p>
</div>
</div>
<div class="section level4">
<h4 id="intersection-analysis-of-de-results">Intersection Analysis of DE Results<a class="anchor" aria-label="anchor" href="#intersection-analysis-of-de-results"></a>
</h4>
<p>Moreover, you can also intersect the DE results of different
normalization methods to see how many DE proteins overlap. You can
either plot for each requested comparison an individual upset plot
(plot_type = “single”) or stack the number of overlapping DE proteins
per comparison (“stacked”). Not only the upset plot(s) are returned, but
also a table with the intersections is provided by the
<code><a href="../reference/plot_upset_DE.html">plot_upset_DE()</a></code>function.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intersections</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_upset_DE.html">plot_upset_DE</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span><span class="op">]</span>, min_degree <span class="op">=</span> <span class="fl">6</span>, plot_type <span class="op">=</span> <span class="st">"stacked"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span>
<span><span class="co"># put legend on top due to very long comparisons</span></span>
<span><span class="va">intersections</span><span class="op">$</span><span class="va">upset</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">intersections</span><span class="op">$</span><span class="va">upset</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"top"</span>, legend.direction <span class="op">=</span> <span class="st">"vertical"</span><span class="op">)</span></span>
<span><span class="va">intersections</span><span class="op">$</span><span class="va">upset</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>Additionally, the Jaccard similarity index can be calculated to
quantify the similarity of the DE results between the different
normalization methods. A individual heatmap can be generated for each
comparison (“plot_type = single”), a single heatmap facetted by
comparison (“facet_comp”) or a single heatmap taking all comparisons
into account (“all”) can be generated.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_jaccard_heatmap.html">plot_jaccard_heatmap</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, plot_type <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Differential_Expression_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>PRONE offers the functionality to extract a consensus set of DEPs
based on a selection of normalization methods and a threshold for the
number of methods that need to agree on the DE status of a protein. The
function <code>get_consensus_DE()</code> returns a list of consensus DE
proteins for either each comparison separately or for all comparisons
combined.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="fu"><a href="../reference/extract_consensus_DE_candidates.html">extract_consensus_DE_candidates</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, norm_thr <span class="op">=</span> <span class="fl">0.8</span>, per_comparison <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64"],["A0A075B6I1","A0A075B6K4;P01715","A0A075B6K5;P80748;A0A075B6K2","A0A075B6K5;P80748;A0A075B6K2","A0A075B730;P58107","A0A0B4J1V6;A0A0B4J1V0","A0A0B4J1Y8","A0A0B4J1Y8","A0A0C4DH34","A0A2R8Y484;Q08722","A0A2R8YDL0;O75443","A0A3B3ISR2;B4DPQ0;P00736;F5H2D0;A0A3B3ITU4;F5H1V0;F5H6Y3;F5H3N3;F5H1N6;F5GWL0;F5H3A3;H0YFL7;Q9NZP8","A1L4H1;M0QZ17","A6NNI4;A0A2R8Y478;A0A7I2V3T0;A0A7I2V3T7;G8JLH6;A0A087WU13;F5GXT1;P21926","B0YIW2;P02656","B0YIW2;P02656","CON__P02663","D6RF35;P02774;D6RBJ7;D6RF20;CON__Q3MHN5;CON__ENSEMBL:ENSBTAP00000018229","F5GXS0;A0A140TA44;A0A0G2JL54;P0C0L5;A0A140TA49;A0A140TA32;A0A140TA29;A0A0G2JPR0;P0C0L4;A0A3B3ISA6;A0A3B3ISD7;CON__P01030;CON__ENSEMBL:ENSBTAP00000007350","H7C4K3;A0A7I2V2F9;E7EUI6;P05556;A0A7I2V348;C9JPK5;H7C4N8;A0A7I2V5Z8;A0A7I2V5T3","J3QQX6;J3QRQ1;P13598;J3QQR8;J3QRT5;J3QKR4","J3QQX6;J3QRQ1;P13598;J3QQR8;J3QRT5;J3QKR4","J3QQX6;J3QRQ1;P13598;J3QQR8;J3QRT5;J3QKR4","K7ER74;P02655;Q6P163;V9GYJ8","K7ER74;P02655;Q6P163;V9GYJ8","K7ERI9;P02654;K7EJI9;K7ELM9;K7EPF9;K7EKP1","K7ERI9;P02654;K7EJI9;K7ELM9;K7EPF9;K7EKP1","O14672","O60832","O60832","O75636","O95445;Q5SRP5","P00738;A0A0C4DGL8;J3QR68;H0Y300;J3QLC9;A0A087WU08;H3BS21;J3KRH2;J3KTC3;J3KSV1;J3QQI8;H3BMJ7","P00738;A0A0C4DGL8;J3QR68;H0Y300;J3QLC9;A0A087WU08;H3BS21;J3KRH2;J3KTC3;J3KSV1;J3QQI8;H3BMJ7","P00738;A0A0C4DGL8;J3QR68;H0Y300;J3QLC9;A0A087WU08;H3BS21;J3KRH2;J3KTC3;J3KSV1;J3QQI8;H3BMJ7","P01009;A0A024R6I7;A0A0G2JRN3;G3V2B9;A0A0G2JPK4;P20848","P01023;CON__ENSEMBL:ENSBTAP00000024146;H0YFH1;F8W7L3","P01699","P01701","P01701","P02452","P02649;H0Y7L5;E9PEV4;E7ERP7;CON__Q03247","P02763","P04004;H0YJW9;F5GX75","P04430","P05106;H3BM21;I3L4X8","P07996;CON__Q28194;A8MZG1;A0A7I2YQ86;A0A7I2V585;A0A7I2YQT6;A0A3B3ITK0;P35442","P08514;A0A3B3IU79","P0DJI8","P0DJI8","P0DJI9","P13501","P27105;F8VSL7","P31947","P55056","P55056","Q04917;F8WEB6;A2IDB1","Q13201;E7EPG1","Q5SZB4","Q68D42","REV__E7EPF5;REV__O43639","REV__H0YI42;REV__Q8IXF0;REV__F8VS42;REV__F8VR32","REV__Q96JF0","REV__Q9Y5T5"],["TBL-HC","TBL-HC","TBL-HC","TBL-PTB","PTB-HC","PTB-HC","TBL-HC","TBL-PTB","TBL-HC","PTB-HC","TBL-PTB","TBL-HC","TBL-HC","PTB-HC","PTB-HC","TBL-HC","Rx-PTB","PTB-HC","PTB-HC","PTB-HC","PTB-HC","TBL-PTB","Rx-PTB","PTB-HC","TBL-HC","PTB-HC","TBL-HC","Rx-PTB","PTB-HC","TBL-HC","Rx-PTB","Rx-PTB","PTB-HC","TBL-PTB","Rx-PTB","PTB-HC","PTB-HC","TBL-PTB","TBL-HC","TBL-PTB","PTB-HC","PTB-HC","PTB-HC","PTB-HC","Rx-PTB","PTB-HC","PTB-HC","PTB-HC","PTB-HC","TBL-PTB","PTB-HC","TBL-HC","PTB-HC","PTB-HC","PTB-HC","TBL-HC","PTB-HC","PTB-HC","TBL-HC","Rx-PTB","Rx-PTB","PTB-HC","Rx-PTB","PTB-HC"],[9,8,8,9,9,8,9,9,8,9,8,9,8,9,8,8,9,9,9,9,9,8,8,8,9,8,8,8,8,8,8,8,9,8,8,9,9,9,9,8,8,8,9,8,8,9,9,9,9,8,9,8,9,9,8,9,9,9,8,8,8,8,9,9],[100,88.88888888888889,88.88888888888889,100,100,88.88888888888889,100,100,88.88888888888889,100,88.88888888888889,100,88.88888888888889,100,88.88888888888889,88.88888888888889,100,100,100,100,100,88.88888888888889,88.88888888888889,88.88888888888889,100,88.88888888888889,88.88888888888889,88.88888888888889,88.88888888888889,88.88888888888889,88.88888888888889,88.88888888888889,100,88.88888888888889,88.88888888888889,100,100,100,100,88.88888888888889,88.88888888888889,88.88888888888889,100,88.88888888888889,88.88888888888889,100,100,100,100,88.88888888888889,100,88.88888888888889,100,100,88.88888888888889,100,100,100,88.88888888888889,88.88888888888889,88.88888888888889,88.88888888888889,100,100]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Protein.IDs<\/th>\n      <th>Comparison<\/th>\n      <th>Intersections<\/th>\n      <th>Percentage<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Protein.IDs","targets":1},{"name":"Comparison","targets":2},{"name":"Intersections","targets":3},{"name":"Percentage","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Luxembourg</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] PRONE_0.99.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rlang_1.1.3                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] GetoptLong_1.0.5            clue_0.3-65                </span></span>
<span><span class="co">#&gt;   [5] matrixStats_1.3.0           compiler_4.4.0             </span></span>
<span><span class="co">#&gt;   [7] reshape2_1.4.4              png_0.1-8                  </span></span>
<span><span class="co">#&gt;   [9] systemfonts_1.1.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [11] stringr_1.5.1               ProtGenerics_1.35.4        </span></span>
<span><span class="co">#&gt;  [13] shape_1.4.6.1               pkgconfig_2.0.3            </span></span>
<span><span class="co">#&gt;  [15] crayon_1.5.2                fastmap_1.1.1              </span></span>
<span><span class="co">#&gt;  [17] XVector_0.43.1              labeling_0.4.3             </span></span>
<span><span class="co">#&gt;  [19] utf8_1.2.4                  rmarkdown_2.26             </span></span>
<span><span class="co">#&gt;  [21] UCSC.utils_0.99.7           preprocessCore_1.65.0      </span></span>
<span><span class="co">#&gt;  [23] ragg_1.3.0                  purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [25] xfun_0.43                   MultiAssayExperiment_1.29.1</span></span>
<span><span class="co">#&gt;  [27] zlibbioc_1.49.3             cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [29] GenomeInfoDb_1.39.14        jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt;  [31] highr_0.10                  DelayedArray_0.29.9        </span></span>
<span><span class="co">#&gt;  [33] BiocParallel_1.37.1         parallel_4.4.0             </span></span>
<span><span class="co">#&gt;  [35] cluster_2.1.6               R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [37] RColorBrewer_1.1-3          stringi_1.8.3              </span></span>
<span><span class="co">#&gt;  [39] bslib_0.7.0                 ComplexUpset_1.3.3         </span></span>
<span><span class="co">#&gt;  [41] limma_3.59.10               GenomicRanges_1.55.4       </span></span>
<span><span class="co">#&gt;  [43] jquerylib_0.1.4             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [45] Rcpp_1.0.12                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [47] knitr_1.46                  IRanges_2.37.1             </span></span>
<span><span class="co">#&gt;  [49] splines_4.4.0               Matrix_1.7-0               </span></span>
<span><span class="co">#&gt;  [51] igraph_2.0.3                tidyselect_1.2.1           </span></span>
<span><span class="co">#&gt;  [53] rstudioapi_0.16.0           abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [55] yaml_2.3.8                  ggtext_0.1.2               </span></span>
<span><span class="co">#&gt;  [57] doParallel_1.0.17           codetools_0.2-20           </span></span>
<span><span class="co">#&gt;  [59] affy_1.81.0                 lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [61] tibble_3.2.1                plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [63] withr_3.0.0                 Biobase_2.63.1             </span></span>
<span><span class="co">#&gt;  [65] evaluate_0.23               desc_1.4.3                 </span></span>
<span><span class="co">#&gt;  [67] xml2_1.3.6                  circlize_0.4.16            </span></span>
<span><span class="co">#&gt;  [69] pillar_1.9.0                affyio_1.73.0              </span></span>
<span><span class="co">#&gt;  [71] BiocManager_1.30.22         MatrixGenerics_1.15.1      </span></span>
<span><span class="co">#&gt;  [73] DT_0.33                     renv_1.0.7                 </span></span>
<span><span class="co">#&gt;  [75] foreach_1.5.2               stats4_4.4.0               </span></span>
<span><span class="co">#&gt;  [77] MSnbase_2.30.1              MALDIquant_1.22.2          </span></span>
<span><span class="co">#&gt;  [79] ncdf4_1.22                  generics_0.1.3             </span></span>
<span><span class="co">#&gt;  [81] S4Vectors_0.42.0            ggplot2_3.5.1              </span></span>
<span><span class="co">#&gt;  [83] munsell_0.5.1               scales_1.3.0               </span></span>
<span><span class="co">#&gt;  [85] glue_1.7.0                  lazyeval_0.2.2             </span></span>
<span><span class="co">#&gt;  [87] tools_4.4.0                 data.table_1.15.4          </span></span>
<span><span class="co">#&gt;  [89] mzID_1.41.0                 QFeatures_1.13.7           </span></span>
<span><span class="co">#&gt;  [91] vsn_3.71.1                  mzR_2.37.3                 </span></span>
<span><span class="co">#&gt;  [93] fs_1.6.4                    XML_3.99-0.16.1            </span></span>
<span><span class="co">#&gt;  [95] grid_4.4.0                  impute_1.77.0              </span></span>
<span><span class="co">#&gt;  [97] tidyr_1.3.1                 crosstalk_1.2.1            </span></span>
<span><span class="co">#&gt;  [99] MsCoreUtils_1.15.7          colorspace_2.1-0           </span></span>
<span><span class="co">#&gt; [101] patchwork_1.2.0             GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt; [103] PSMatch_1.7.2               cli_3.6.2                  </span></span>
<span><span class="co">#&gt; [105] textshaping_0.3.7           fansi_1.0.6                </span></span>
<span><span class="co">#&gt; [107] S4Arrays_1.3.7              ComplexHeatmap_2.20.0      </span></span>
<span><span class="co">#&gt; [109] dplyr_1.1.4                 AnnotationFilter_1.27.0    </span></span>
<span><span class="co">#&gt; [111] pcaMethods_1.95.0           gtable_0.3.5               </span></span>
<span><span class="co">#&gt; [113] sass_0.4.9                  digest_0.6.35              </span></span>
<span><span class="co">#&gt; [115] BiocGenerics_0.49.1         SparseArray_1.3.5          </span></span>
<span><span class="co">#&gt; [117] rjson_0.2.21                farver_2.1.1               </span></span>
<span><span class="co">#&gt; [119] htmlwidgets_1.6.4           memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [121] htmltools_0.5.8.1           pkgdown_2.0.9              </span></span>
<span><span class="co">#&gt; [123] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [125] GlobalOptions_0.1.2         statmod_1.5.0              </span></span>
<span><span class="co">#&gt; [127] gridtext_0.1.5              MASS_7.3-60.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg
Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen
Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium
Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes
in Infected Patients Depending on Tuberculosis Disease State.”</span>
<em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
<div id="ref-ritchie_limma_2015" class="csl-entry">
Ritchie, Matthew E., Belinda Phipson, Di Wu, Yifang Hu, Charity W. Law,
Wei Shi, and Gordon K. Smyth. 2015. <span>“Limma Powers Differential
Expression Analyses for Rna-Sequencing and Microarray Studies.”</span>
<em>Nucleic Acids Research</em> 43 (7): e47–47. <a href="https://doi.org/10.1093/nar/gkv007" class="external-link">https://doi.org/10.1093/nar/gkv007</a>.
</div>
<div id="ref-suomi_rots:_2017" class="csl-entry">
Suomi, Tomi, Fatemeh Seyednasrollah, Maria K. Jaakkola, Thomas Faux, and
Laura L. Elo. 2017. <span>“<span>ROTS</span>: <span>An</span>
<span>R</span> Package for Reproducibility-Optimized Statistical
Testing.”</span> Edited by Timothée Poisot. <em>PLOS Computational
Biology</em> 13 (5): e1005562. <a href="https://doi.org/10.1371/journal.pcbi.1005562" class="external-link">https://doi.org/10.1371/journal.pcbi.1005562</a>.
</div>
<div id="ref-zhu_deqms:_2020" class="csl-entry">
Zhu, Yafeng, Lukas M. Orre, Yan Zhou Tran, Georgios Mermelekas, Henrik
J. Johansson, Alina Malyutina, Simon Anders, and Janne Lehtiö. 2020.
<span>“Deqms: A Method for Accurate Variance Estimation in Differential
Protein Expression Analysis.”</span> <em>Molecular &amp; Cellular
Proteomics</em> 19 (6): 1047–57. <a href="https://doi.org/10.1074/mcp.TIR119.001646" class="external-link">https://doi.org/10.1074/mcp.TIR119.001646</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
