<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PRONE">
<title>Imputation • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Imputation">
<meta property="og:description" content="PRONE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/PRONE.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/Normalization.html">Normalization</a>
    <a class="dropdown-item" href="../articles/Imputation.html">Imputation</a>
    <a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a>
    <a class="dropdown-item" href="../articles/Spike_In_Data.html">How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Imputation</h1>
                        <h4 data-toc-skip class="author">Arend Lis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Imputation.Rmd" class="external-link"><code>vignettes/Imputation.Rmd</code></a></small>
      <div class="d-none name"><code>Imputation.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="load-data-tmt">Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a>
data. For more information on how to create the SummarizedExperiment
from a proteomics data set, please refer to the <a href="PRONE.Rmd">“Get
Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>As we have seen in the Preprocessing phase, that samples “1.HC_Pool1”
and “1.HC_Pool2” have been removed from the data set due to their high
amount of missing values (more than 80% of NAs per sample), before
imputing the data we will here remove these two samples.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples_manually.html">remove_samples_manually</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Label"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.HC_Pool1"</span>, <span class="st">"1.HC_Pool2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 samples removed.</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="impute-data">Impute Data<a class="anchor" aria-label="anchor" href="#impute-data"></a>
</h2>
<p>Currently, there is only a mixed imputation method available in
PRONE: k-nearest neighbor imputation for proteins with missing values at
random and a left-shifted Gaussian distribution for proteins with
missing values not at random. Imputation can be performed on a selection
of normalized data sets using the “ain” parameter in the
<code>impute_SE</code> function. The default is to impute all assays
(ain = NULL).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute_se.html">impute_se</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span>
<span><span class="co">#&gt; Imputing along margin 1 (features/rows).</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Luxembourg</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] PRONE_0.99.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rlang_1.1.3                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] clue_0.3-65                 matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt;   [5] compiler_4.4.0              systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;   [7] vctrs_0.6.5                 ProtGenerics_1.35.4        </span></span>
<span><span class="co">#&gt;   [9] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [11] fastmap_1.1.1               XVector_0.43.1             </span></span>
<span><span class="co">#&gt;  [13] utf8_1.2.4                  rmarkdown_2.26             </span></span>
<span><span class="co">#&gt;  [15] UCSC.utils_0.99.7           preprocessCore_1.65.0      </span></span>
<span><span class="co">#&gt;  [17] ragg_1.3.0                  purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [19] xfun_0.43                   MultiAssayExperiment_1.29.1</span></span>
<span><span class="co">#&gt;  [21] zlibbioc_1.49.3             cachem_1.0.8               </span></span>
<span><span class="co">#&gt;  [23] GenomeInfoDb_1.39.14        jsonlite_1.8.8             </span></span>
<span><span class="co">#&gt;  [25] DelayedArray_0.29.9         BiocParallel_1.37.1        </span></span>
<span><span class="co">#&gt;  [27] parallel_4.4.0              cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [29] R6_2.5.1                    bslib_0.7.0                </span></span>
<span><span class="co">#&gt;  [31] limma_3.59.10               GenomicRanges_1.55.4       </span></span>
<span><span class="co">#&gt;  [33] jquerylib_0.1.4             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [35] Rcpp_1.0.12                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">#&gt;  [37] knitr_1.46                  IRanges_2.37.1             </span></span>
<span><span class="co">#&gt;  [39] Matrix_1.7-0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt;  [41] tidyselect_1.2.1            rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [43] abind_1.4-5                 yaml_2.3.8                 </span></span>
<span><span class="co">#&gt;  [45] ggtext_0.1.2                doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [47] codetools_0.2-20            affy_1.81.0                </span></span>
<span><span class="co">#&gt;  [49] lattice_0.22-6              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [51] plyr_1.8.9                  withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [53] Biobase_2.63.1              evaluate_0.23              </span></span>
<span><span class="co">#&gt;  [55] desc_1.4.3                  xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [57] pillar_1.9.0                affyio_1.73.0              </span></span>
<span><span class="co">#&gt;  [59] BiocManager_1.30.22         MatrixGenerics_1.15.1      </span></span>
<span><span class="co">#&gt;  [61] renv_1.0.7                  foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [63] stats4_4.4.0                MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt;  [65] MALDIquant_1.22.2           ncdf4_1.22                 </span></span>
<span><span class="co">#&gt;  [67] generics_0.1.3              S4Vectors_0.42.0           </span></span>
<span><span class="co">#&gt;  [69] ggplot2_3.5.1               munsell_0.5.1              </span></span>
<span><span class="co">#&gt;  [71] scales_1.3.0                glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [73] lazyeval_0.2.2              tools_4.4.0                </span></span>
<span><span class="co">#&gt;  [75] data.table_1.15.4           mzID_1.41.0                </span></span>
<span><span class="co">#&gt;  [77] QFeatures_1.13.7            vsn_3.71.1                 </span></span>
<span><span class="co">#&gt;  [79] mzR_2.37.3                  fs_1.6.4                   </span></span>
<span><span class="co">#&gt;  [81] XML_3.99-0.16.1             grid_4.4.0                 </span></span>
<span><span class="co">#&gt;  [83] impute_1.77.0               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [85] MsCoreUtils_1.15.7          colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [87] GenomeInfoDbData_1.2.12     PSMatch_1.7.2              </span></span>
<span><span class="co">#&gt;  [89] cli_3.6.2                   textshaping_0.3.7          </span></span>
<span><span class="co">#&gt;  [91] fansi_1.0.6                 S4Arrays_1.3.7             </span></span>
<span><span class="co">#&gt;  [93] dplyr_1.1.4                 AnnotationFilter_1.27.0    </span></span>
<span><span class="co">#&gt;  [95] pcaMethods_1.95.0           gtable_0.3.5               </span></span>
<span><span class="co">#&gt;  [97] sass_0.4.9                  digest_0.6.35              </span></span>
<span><span class="co">#&gt;  [99] BiocGenerics_0.49.1         SparseArray_1.3.5          </span></span>
<span><span class="co">#&gt; [101] htmlwidgets_1.6.4           memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [103] htmltools_0.5.8.1           pkgdown_2.0.9              </span></span>
<span><span class="co">#&gt; [105] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [107] statmod_1.5.0               gridtext_0.1.5             </span></span>
<span><span class="co">#&gt; [109] MASS_7.3-60.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg
Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen
Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium
Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes
in Infected Patients Depending on Tuberculosis Disease State.”</span>
<em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
