<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PRONE">
<title>Preprocessing • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Preprocessing">
<meta property="og:description" content="PRONE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/PRONE.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/Normalization.html">Normalization</a>
    <a class="dropdown-item" href="../articles/Imputation.html">Imputation</a>
    <a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a>
    <a class="dropdown-item" href="../articles/Spike_In_Data.html">How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<link href="Preprocessing_files/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="Preprocessing_files/htmlwidgets-1.6.4/htmlwidgets.js"></script><link href="Preprocessing_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="Preprocessing_files/datatables-binding-0.33/datatables.js"></script><link href="Preprocessing_files/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="Preprocessing_files/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="Preprocessing_files/dt-core-1.13.6/js/jquery.dataTables.min.js"></script><link href="Preprocessing_files/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="Preprocessing_files/crosstalk-1.2.1/js/crosstalk.min.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Preprocessing</h1>
                        <h4 data-toc-skip class="author">Arend Lis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Preprocessing.Rmd" class="external-link"><code>vignettes/Preprocessing.Rmd</code></a></small>
      <div class="d-none name"><code>Preprocessing.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="load-data-tmt">Load Data (TMT)<a class="anchor" aria-label="anchor" href="#load-data-tmt"></a>
</h2>
<p>Here, we are directly working with the <a href="https://bioconductor.org/packages/release/bioc/html/SummarizedExperiment.html" class="external-link">SummarizedExperiment</a>
data. For more information on how to create the SummarizedExperiment
from a proteomics data set, please refer to the <a href="PRONE.html">“Get Started”</a> vignette.</p>
<p>The example TMT data set originates from <span class="citation">(Biadglegne et al. 2022)</span>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tuberculosis_TMT_se"</span><span class="op">)</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="va">tuberculosis_TMT_se</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="overview-of-the-data">Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h2>
<p>To get an overview on the number of NAs, you can simply use the
function <code><a href="../reference/get_NA_overview.html">get_NA_overview()</a></code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_NA_overview.html">get_NA_overview</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Total.Values NA.Values NA.Percentage</span></span>
<span><span class="co">#&gt;           &lt;int&gt;     &lt;int&gt;         &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1:         6020      1945      32.30897</span></span></code></pre></div>
<p>To get an overview on the number of samples per sample group or
batch, you can simply use the function
<code><a href="../reference/plot_condition_overview.html">plot_condition_overview()</a></code> by specifying the column of the
meta-data that should be used for coloring. By default (condition =
NULL), the column specified in <code><a href="../reference/load_data.html">load_data()</a></code>will be used.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/overview_plots-1.png" width="700"></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="st">"Pool"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/overview_plots-2.png" width="700"></p>
<p>A general overview of the protein intensities across the different
samples is provided by the function <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>. The
parameter “ain” specifies the data to plot, currently only “raw” and
“log2” is available (names(assays(se)). Later if multiple normalization
methods are executed, these will be saved as assays, and the normalized
data can be visualized.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">available_ains</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, color_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pool"</span>, <span class="st">"Group"</span><span class="op">)</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $log2</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/overview_heatmap-1.png" width="700"></p>
<p>Similarly, an upset plot can be generated to visualize the overlaps
between sets defined by a specific column in the metadata. The sets are
generated by using non-NA values.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, mb.ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>,<span class="fl">0.3</span><span class="op">)</span>, only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/overview_upset-1.png" width="768"></p>
<p>TODO: plot_pairs_panels !</p>
<p>If you are interested in the intensities of specific biomarkers, you
can use the <code><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots()</a></code> function to compare the
distribution of intensities per group. The plot can be generated per
marker and facet by normalization method (facet_norm = TRUE) or by
normalization method and facet by marker (facet_marker = TRUE).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots</a></span><span class="op">(</span><span class="va">se</span>, markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q92954;J3KP74;E9PLR3"</span>, <span class="st">"Q9Y6Z7"</span>, <span class="st">"Q68CQ4"</span><span class="op">)</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, id_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span>, facet_marker <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; No shaping done.</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="filter-proteins">Filter Proteins<a class="anchor" aria-label="anchor" href="#filter-proteins"></a>
</h2>
<div class="section level3">
<h3 id="remove-proteins-with-missing-values-in-all-samples">Remove Proteins With Missing Values in ALL Samples<a class="anchor" aria-label="anchor" href="#remove-proteins-with-missing-values-in-all-samples"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_complete_NA_proteins.html">filter_out_complete_NA_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 13 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-proteins-with-a-specific-value-in-a-specific-column">Remove Proteins With a Specific Value in a Specific Column<a class="anchor" aria-label="anchor" href="#remove-proteins-with-a-specific-value-in-a-specific-column"></a>
</h3>
<p>Typically proteins with “+” in the columns “Reverse”,
“Only.identified.by.site”, and “Potential.contaminant” are removed in
case of a MaxQuant proteinGroups.txt output file.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_value.html">filter_out_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Reverse"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 17 proteins were removed.</span></span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_value.html">filter_out_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Only.identified.by.site"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 proteins were removed.</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#se &lt;- filter_out_proteins_by_value(se, "Potential.contaminant", "+")</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-proteins-by-id">Remove Proteins by ID<a class="anchor" aria-label="anchor" href="#remove-proteins-by-id"></a>
</h3>
<p>If you don’t want to remove for instance all proteins with
“Potential.contaminant == +”, you can also first get the protein ID with
the specific value, check them in Uniprot, and then remove only some by
using the function <code><a href="../reference/filter_out_proteins_by_ID.html">filter_out_proteins_by_ID()</a></code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pot_contaminants</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_proteins_by_value.html">get_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Potential.contaminant"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were identified.</span></span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_proteins_by_ID.html">filter_out_proteins_by_ID</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">pot_contaminants</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="explore-missing-value-pattern">Explore Missing Value Pattern<a class="anchor" aria-label="anchor" href="#explore-missing-value-pattern"></a>
</h3>
<p>Due to the high amount of missing values in MS-based proteomics data,
it is important to explore the missing value pattern in the data. The
function <code><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap()</a></code> provides a heatmap of the
proteins with at least one missing value across all samples.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, cluster_samples <span class="op">=</span> <span class="cn">TRUE</span>, cluster_proteins <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Another way to explore the missing value pattern is to use the
functions <code><a href="../reference/plot_NA_density.html">plot_NA_density()</a></code> and
<code><a href="../reference/plot_NA_frequency.html">plot_NA_frequency()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_density.html">plot_NA_density</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_frequency.html">plot_NA_frequency</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="filter-proteins-by-applying-a-missing-value-threshold">Filter Proteins By Applying a Missing Value Threshold<a class="anchor" aria-label="anchor" href="#filter-proteins-by-applying-a-missing-value-threshold"></a>
</h3>
<p>To reduce the amount of missing values, it is possible to filter
proteins by applying a missing value threshold. The function
<code><a href="../reference/filter_out_NA_proteins_by_threshold.html">filter_out_NA_proteins_by_threshold()</a></code> removes proteins with
more missing values than the specified threshold. The threshold is a
value between 0 and 1, where 0.7, for instance, means that proteins with
less than 70% of real values will be removed, i.e., proteins with more
than 30% missing values will be removed.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_out_NA_proteins_by_threshold.html">filter_out_NA_proteins_by_threshold</a></span><span class="op">(</span><span class="va">se</span>, thr <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> </span>
<span><span class="co">#&gt; 99 proteins were removed.</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="filter-samples">Filter Samples<a class="anchor" aria-label="anchor" href="#filter-samples"></a>
</h2>
<p>Following filtering proteins by different criteria, samples can be
analyzed more in detail. PRONE provides some functions, such as
<code><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples()</a></code> and
<code><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples()</a></code>, to get an overview of the number of
proteins and the total intensity per sample, but also offers the
automatic outlier detection method of POMA.</p>
<div class="section level3">
<h3 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h3>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-11-2.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="remove-samples-manually">Remove Samples Manually<a class="anchor" aria-label="anchor" href="#remove-samples-manually"></a>
</h3>
<p>Based on these plots, samples “1.HC_Pool1” and 1_HC_Pool2 seem to be
outliers. You can easily remove samples manually by using the
<code><a href="../reference/remove_samples_manually.html">remove_samples_manually()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples_manually.html">remove_samples_manually</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Label"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.HC_Pool1"</span>, <span class="st">"1.HC_Pool2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 samples removed.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="remove-reference-samples">Remove Reference Samples<a class="anchor" aria-label="anchor" href="#remove-reference-samples"></a>
</h3>
<p>And you can remove the reference samples directly using the function
<code><a href="../reference/remove_reference_samples.html">remove_reference_samples()</a></code>. But attention: possibly you
need them for normalization! That is exactly why we currently keep
them!</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_no_refs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_reference_samples.html">remove_reference_samples</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 reference samples removed from the SummarizedExperiment object.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="outlier-detection-via-poma-r-package">Outlier Detection via POMA R Package<a class="anchor" aria-label="anchor" href="#outlier-detection-via-poma-r-package"></a>
</h3>
<p>The POMA R package provides a method to detect outliers in proteomics
data. The function <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> detects outliers
in the data based on the POMA algorithm. The function returns a list
with the following elements: polygon plot, distance boxplot, and the
outliers. For further information on the POMA algorithm, please refer to
the original publication <span class="citation">(Castellano-Escuder et
al. 2021)</span>:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">poma_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">poma_res</span><span class="op">$</span><span class="va">polygon_plot</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">poma_res</span><span class="op">$</span><span class="va">distance_boxplot</span></span></code></pre></div>
<p><img src="Preprocessing_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu">DT</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/DT/man/datatable.html" class="external-link">datatable</a></span><span class="op">(</span><span class="va">poma_res</span><span class="op">$</span><span class="va">outliers</span>, options <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>scrollX <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ac96cb3ee4656e2e9ec3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ac96cb3ee4656e2e9ec3">{"x":{"filter":"none","vertical":false,"data":[["1"],["Reporter.intensity.corrected.10.Pat_tbc_pool_2"],["PTB"],[11.04724464907511],[10.52433157277052]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>sample<\/th>\n      <th>groups<\/th>\n      <th>distance_to_centroid<\/th>\n      <th>limit_distance<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"columnDefs":[{"className":"dt-right","targets":[3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"sample","targets":1},{"name":"groups","targets":2},{"name":"distance_to_centroid","targets":3},{"name":"limit_distance","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script><p>To remove the outliers detected via the POMA algorithm, just put the
data.table of the <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> function into the
<code><a href="../reference/remove_POMA_outliers.html">remove_POMA_outliers()</a></code> function.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_POMA_outliers.html">remove_POMA_outliers</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">poma_res</span><span class="op">$</span><span class="va">outliers</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 outlier samples removed.</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] PRONE_0.99.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] RColorBrewer_1.1-3          rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;   [3] jsonlite_1.8.8              shape_1.4.6.1              </span></span>
<span><span class="co">#&gt;   [5] MultiAssayExperiment_1.30.1 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [7] magick_2.8.3                farver_2.1.2               </span></span>
<span><span class="co">#&gt;   [9] MALDIquant_1.22.2           rmarkdown_2.27             </span></span>
<span><span class="co">#&gt;  [11] GlobalOptions_0.1.2         fs_1.6.4                   </span></span>
<span><span class="co">#&gt;  [13] zlibbioc_1.50.0             ragg_1.3.2                 </span></span>
<span><span class="co">#&gt;  [15] vctrs_0.6.5                 memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [17] janitor_2.2.0               htmltools_0.5.8.1          </span></span>
<span><span class="co">#&gt;  [19] S4Arrays_1.4.1              SparseArray_1.4.8          </span></span>
<span><span class="co">#&gt;  [21] mzID_1.42.0                 sass_0.4.9                 </span></span>
<span><span class="co">#&gt;  [23] bslib_0.7.0                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt;  [25] desc_1.4.3                  plyr_1.8.9                 </span></span>
<span><span class="co">#&gt;  [27] lubridate_1.9.3             impute_1.78.0              </span></span>
<span><span class="co">#&gt;  [29] cachem_1.1.0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt;  [31] lifecycle_1.0.4             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [33] pkgconfig_2.0.3             Matrix_1.7-0               </span></span>
<span><span class="co">#&gt;  [35] R6_2.5.1                    fastmap_1.2.0              </span></span>
<span><span class="co">#&gt;  [37] snakecase_0.11.1            GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">#&gt;  [39] MatrixGenerics_1.16.0       clue_0.3-65                </span></span>
<span><span class="co">#&gt;  [41] digest_0.6.35               pcaMethods_1.96.0          </span></span>
<span><span class="co">#&gt;  [43] colorspace_2.1-0            S4Vectors_0.42.0           </span></span>
<span><span class="co">#&gt;  [45] crosstalk_1.2.1             textshaping_0.4.0          </span></span>
<span><span class="co">#&gt;  [47] GenomicRanges_1.56.0        vegan_2.6-6.1              </span></span>
<span><span class="co">#&gt;  [49] labeling_0.4.3              timechange_0.3.0           </span></span>
<span><span class="co">#&gt;  [51] fansi_1.0.6                 mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [53] httr_1.4.7                  abind_1.4-5                </span></span>
<span><span class="co">#&gt;  [55] compiler_4.4.0              withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [57] doParallel_1.0.17           BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt;  [59] UpSetR_1.4.0                highr_0.11                 </span></span>
<span><span class="co">#&gt;  [61] MASS_7.3-60.2               DelayedArray_0.30.1        </span></span>
<span><span class="co">#&gt;  [63] rjson_0.2.21                permute_0.9-7              </span></span>
<span><span class="co">#&gt;  [65] mzR_2.38.0                  tools_4.4.0                </span></span>
<span><span class="co">#&gt;  [67] PSMatch_1.8.0               glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [69] nlme_3.1-164                QFeatures_1.14.1           </span></span>
<span><span class="co">#&gt;  [71] gridtext_0.1.5              grid_4.4.0                 </span></span>
<span><span class="co">#&gt;  [73] cluster_2.1.6               reshape2_1.4.4             </span></span>
<span><span class="co">#&gt;  [75] generics_0.1.3              gtable_0.3.5               </span></span>
<span><span class="co">#&gt;  [77] preprocessCore_1.66.0       tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [79] data.table_1.15.4           xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [81] utf8_1.2.4                  XVector_0.44.0             </span></span>
<span><span class="co">#&gt;  [83] BiocGenerics_0.50.0         foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [85] pillar_1.9.0                stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [87] limma_3.60.2                splines_4.4.0              </span></span>
<span><span class="co">#&gt;  [89] circlize_0.4.16             dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [91] ggtext_0.1.2                lattice_0.22-6             </span></span>
<span><span class="co">#&gt;  [93] tidyselect_1.2.1            ComplexHeatmap_2.20.0      </span></span>
<span><span class="co">#&gt;  [95] knitr_1.47                  gridExtra_2.3              </span></span>
<span><span class="co">#&gt;  [97] IRanges_2.38.0              ProtGenerics_1.36.0        </span></span>
<span><span class="co">#&gt;  [99] SummarizedExperiment_1.34.0 stats4_4.4.0               </span></span>
<span><span class="co">#&gt; [101] xfun_0.45                   Biobase_2.64.0             </span></span>
<span><span class="co">#&gt; [103] statmod_1.5.0               MSnbase_2.30.1             </span></span>
<span><span class="co">#&gt; [105] matrixStats_1.3.0           DT_0.33                    </span></span>
<span><span class="co">#&gt; [107] stringi_1.8.4               UCSC.utils_1.0.0           </span></span>
<span><span class="co">#&gt; [109] lazyeval_0.2.2              yaml_2.3.8                 </span></span>
<span><span class="co">#&gt; [111] evaluate_0.24.0             codetools_0.2-20           </span></span>
<span><span class="co">#&gt; [113] MsCoreUtils_1.16.0          tibble_3.2.1               </span></span>
<span><span class="co">#&gt; [115] BiocManager_1.30.23         cli_3.6.2                  </span></span>
<span><span class="co">#&gt; [117] affyio_1.74.0               systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt; [119] munsell_0.5.1               jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt; [121] Rcpp_1.0.12                 GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt; [123] png_0.1-8                   XML_3.99-0.16.1            </span></span>
<span><span class="co">#&gt; [125] parallel_4.4.0              pkgdown_2.0.9              </span></span>
<span><span class="co">#&gt; [127] ggplot2_3.5.1               dendsort_0.3.4             </span></span>
<span><span class="co">#&gt; [129] AnnotationFilter_1.28.0     scales_1.3.0               </span></span>
<span><span class="co">#&gt; [131] affy_1.82.0                 ncdf4_1.22                 </span></span>
<span><span class="co">#&gt; [133] purrr_1.0.2                 crayon_1.5.2               </span></span>
<span><span class="co">#&gt; [135] POMA_1.14.0                 GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt; [137] rlang_1.1.4                 vsn_3.72.0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-biadglegne_mycobacterium_2022" class="csl-entry">
Biadglegne, Fantahun, Johannes R. Schmidt, Kathrin M. Engel, Jörg
Lehmann, Robert T. Lehmann, Anja Reinert, Brigitte König, Jürgen
Schiller, Stefan Kalkhof, and Ulrich Sack. 2022. <span>“Mycobacterium
Tuberculosis Affects Protein and Lipid Content of Circulating Exosomes
in Infected Patients Depending on Tuberculosis Disease State.”</span>
<em>Biomedicines</em> 10 (4): 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link">https://doi.org/10.3390/biomedicines10040783</a>.
</div>
<div id="ref-castellano-escuder_pomashiny:_2021" class="csl-entry">
Castellano-Escuder, Pol, Raúl González-Domínguez, Francesc
Carmona-Pontaque, Cristina Andrés-Lacueva, and Alex Sánchez-Pla. 2021.
<span>“<span>POMAShiny</span>: <span>A</span> User-Friendly Web-Based
Workflow for Metabolomics and Proteomics Data Analysis.”</span> Edited
by Manja Marz. <em>PLOS Computational Biology</em> 17 (7): e1009148. <a href="https://doi.org/10.1371/journal.pcbi.1009148" class="external-link">https://doi.org/10.1371/journal.pcbi.1009148</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
