<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PRONE.R">
<title>Getting started with PRONE • PRONE.R</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with PRONE">
<meta property="og:description" content="PRONE.R">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PRONE.R</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.25</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/PRONE.R.html">Getting started with PRONE</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lisiarend/PRONE.R/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Getting started with PRONE</h1>
                        <h4 data-toc-skip class="author">Arend Lis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE.R/blob/HEAD/vignettes/PRONE.R.Rmd" class="external-link"><code>vignettes/PRONE.R.Rmd</code></a></small>
      <div class="d-none name"><code>PRONE.R.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install PRONE.R from github and build vignettes</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"lisiarend/PRONE.R"</span><span class="op">)</span></span>
<span><span class="co"># Load and attach PRONE.R </span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lisiarend/PRONE.R" class="external-link">"PRONE.R"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="real-world-data-sets">Real-world Data Sets<a class="anchor" aria-label="anchor" href="#real-world-data-sets"></a>
</h2>
<p>The following sections present the functionalities of PRONE based on
a TMT and LFQ data set.</p>
<p>The example TMT data set originates of:</p>
<p>Fantahun Biadglegne et al. Mycobacterium tuberculosis Affects Protein
and Lipid Content of Circulating Exosomes in Infected Patients Depending
on Tuberculosis Disease State. Biomedicines 10.4 (Mar. 2022), p. 783. <a href="https://doi.org/10.3390/biomedicines10040783" class="external-link uri">https://doi.org/10.3390/biomedicines10040783</a>.</p>
<p>The example LFQ data set originates of:</p>
<p>Anni P. Vehmas et al. Liver lipid metabolism is altered by increased
circulating estrogen to androgen ratio in male mouse. Journal of
Proteomics 133 (Feb. 2016), pp. 66–75. <a href="https://doi.org/10.1016/j.jprot.2015.12.009" class="external-link uri">https://doi.org/10.1016/j.jprot.2015.12.009</a>.</p>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>For loading the data, you need to first load the protein intensity
table and meta-data information in a data frame If you have a TMT data
set with reference samples, you need to specify the reference samples.
Then, you can call the <code><a href="../reference/load_data.html">load_data()</a></code> function to load the
data into a SummarizedExperiment by specifying some additional
parameters. You need to make sure that the sample names are saved in a
column named “Column” in the meta-data table and are named accordingly
in the protein intensity table.</p>
<div class="section level4">
<h4 id="example-1-tmt-data-set">Example 1: TMT Data Set<a class="anchor" aria-label="anchor" href="#example-1-tmt-data-set"></a>
</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../inst/extdata/real_world/TMT/proteinGroups.txt"</span>, sep <span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../inst/extdata/real_world/TMT/metadata.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">md</span><span class="op">$</span><span class="va">Column</span> <span class="op">&lt;-</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">md</span><span class="op">$</span><span class="va">Column</span>, <span class="st">" "</span>, <span class="st">"."</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ref_samples</span> <span class="op">&lt;-</span> <span class="va">md</span><span class="op">[</span><span class="va">md</span><span class="op">$</span><span class="va">Group</span> <span class="op">==</span> <span class="st">"Common.reference"</span>,<span class="op">]</span><span class="op">$</span><span class="va">Column</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">md</span>, protein_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, gene_column <span class="op">=</span> <span class="st">"Gene.names"</span>, ref_samples <span class="op">=</span> <span class="va">ref_samples</span>, batch_column <span class="op">=</span> <span class="st">"Pool"</span>, condition_column <span class="op">=</span> <span class="st">"Group"</span>, label_column <span class="op">=</span> <span class="st">"Label"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-2-lfq-data-set">Example 2: LFQ Data Set<a class="anchor" aria-label="anchor" href="#example-2-lfq-data-set"></a>
</h4>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">LFQ_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../inst/extdata/real_world/LFQ/MOUSE-Data-NonNormalized.csv"</span>, row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create metadata (since it was not given)</span></span>
<span><span class="va">LFQ_md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">LFQ_data</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">LFQ_data</span><span class="op">)</span><span class="op">-</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="va">LFQ_md</span><span class="op">$</span><span class="va">Animal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">LFQ_md</span><span class="op">$</span><span class="va">Column</span><span class="op">)</span>,<span class="st">"_"</span><span class="op">)</span>, <span class="st">"["</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">LFQ_md</span><span class="op">$</span><span class="va">Condition</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="va">LFQ_md</span><span class="op">$</span><span class="va">Column</span><span class="op">)</span>,<span class="st">"_"</span><span class="op">)</span>, <span class="st">"["</span>, <span class="fl">2</span><span class="op">)</span></span>
<span></span>
<span><span class="va">LFQ_se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_data.html">load_data</a></span><span class="op">(</span><span class="va">LFQ_data</span>, <span class="va">LFQ_md</span>, protein_column <span class="op">=</span> <span class="st">"Accession"</span>, gene_column <span class="op">=</span> <span class="cn">NULL</span>, ref_samples <span class="op">=</span> <span class="cn">NULL</span>, batch_column <span class="op">=</span> <span class="cn">NULL</span>, condition_column <span class="op">=</span> <span class="st">"Condition"</span>, label_column <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="overview-of-the-data">Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h3>
<p>To get an overview on the number of NAs, you can simply use the
function <code><a href="../reference/get_NA_overview.html">get_NA_overview()</a></code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_NA_overview.html">get_NA_overview</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Total.Values NA.Values NA.Percentage</span></span>
<span><span class="co">#&gt;           &lt;int&gt;     &lt;int&gt;         &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1:         6020      1945      32.30897</span></span></code></pre></div>
<p>To get an overview on the number of samples per sample group or
batch, you can simply use the function
<code><a href="../reference/plot_condition_overview.html">plot_condition_overview()</a></code> by specifying the column of the
meta-data that should be used for coloring. By default (condition =
NULL), the column specified in <code><a href="../reference/load_data.html">load_data()</a></code>will be used.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_condition_overview.html">plot_condition_overview</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="st">"Pool"</span><span class="op">)</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-6-2.png" width="700"></p>
<p>A general overview of the protein intensities across the different
samples is provided by the function <code><a href="../reference/plot_heatmap.html">plot_heatmap()</a></code>. The
parameter “ain” specifies the data to plot, currently only “raw” and
“log2” is available (names(assays(se)). Later if multiple normalization
methods are executed, these will be saved as assays, and the normalized
data can be visualized.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">available_ains</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span><span class="va">se</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, color_by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Pool"</span>, <span class="st">"Group"</span><span class="op">)</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $log2</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>Similarly, an upset plot can be generated to visualize the overlaps
between sets defined by a specific column in the metadata. The sets are
generated by using non-NA values.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_upset.html">plot_upset</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, mb.ratio <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.7</span>,<span class="fl">0.3</span><span class="op">)</span>, only_refs <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>TODO: plot_pairs_panels !</p>
<p>If you are interested in the intensities of specific biomarkers, you
can use the <code><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots()</a></code> function to compare the
distribution of intensities per group. The plot can be generated per
marker and facet by normalization method (facet_norm = TRUE) or by
normalization method and facet by marker (facet_marker = TRUE).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_markers_boxplots.html">plot_markers_boxplots</a></span><span class="op">(</span><span class="va">se</span>, markers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Q92954;J3KP74;E9PLR3"</span>, <span class="st">"Q9Y6Z7"</span>, <span class="st">"Q68CQ4"</span><span class="op">)</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, id_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span>, facet_marker <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; No shaping done.</span></span>
<span><span class="va">p</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">90</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 class="unnumbered" id="filter-proteins">Filter Proteins<a class="anchor" aria-label="anchor" href="#filter-proteins"></a>
</h3>
<div class="section level4">
<h4 id="remove-proteins-with-missing-values-in-all-samples">Remove Proteins With Missing Values in ALL Samples<a class="anchor" aria-label="anchor" href="#remove-proteins-with-missing-values-in-all-samples"></a>
</h4>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_complete_NA_proteins.html">filter_complete_NA_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 13 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="remove-proteins-with-a-specific-value-in-a-specific-column">Remove Proteins With a Specific Value in a Specific Column<a class="anchor" aria-label="anchor" href="#remove-proteins-with-a-specific-value-in-a-specific-column"></a>
</h4>
<p>Typically proteins with “+” in the columns “Reverse”,
“Only.identified.by.site”, and “Potential.contaminant” are removed in
case of a MaxQuant proteinGroups.txt output file.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_proteins_by_value.html">filter_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Reverse"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 17 proteins were removed.</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_proteins_by_value.html">filter_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Only.identified.by.site"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 1 proteins were removed.</span></span>
<span><span class="co">#se &lt;- filter_proteins_by_value(se, "Potential.contaminant", "+")</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="remove-proteins-by-id">Remove Proteins by ID<a class="anchor" aria-label="anchor" href="#remove-proteins-by-id"></a>
</h4>
<p>If you don’t want to remove for instance all proteins with
“Potential.contaminant == +”, you can also first get the protein ID with
the specific value, check them in Uniprot, and then remove only some by
using the function <code><a href="../reference/filter_proteins_by_ID.html">filter_proteins_by_ID()</a></code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pot_contaminants</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_proteins_by_value.html">get_proteins_by_value</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Potential.contaminant"</span>, <span class="st">"+"</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were identified.</span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_proteins_by_ID.html">filter_proteins_by_ID</a></span><span class="op">(</span><span class="va">se</span>, <span class="va">pot_contaminants</span><span class="op">)</span></span>
<span><span class="co">#&gt; 24 proteins were removed.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="explore-missing-value-pattern">Explore Missing Value Pattern<a class="anchor" aria-label="anchor" href="#explore-missing-value-pattern"></a>
</h4>
<p>Due to the high amount of missing values in MS-based proteomics data,
it is important to explore the missing value pattern in the data. The
function <code><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap()</a></code> provides a heatmap of the
proteins with at least one missing value across all samples.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, cluster_samples <span class="op">=</span> <span class="cn">TRUE</span>, cluster_proteins <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-13-1.png" width="700">
Another way to explore the missing value pattern is to use the functions
<code><a href="../reference/plot_NA_density.html">plot_NA_density()</a></code> and <code><a href="../reference/plot_NA_frequency.html">plot_NA_frequency()</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_density.html">plot_NA_density</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_NA_frequency.html">plot_NA_frequency</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="filter-proteins-by-applying-a-missing-value-threshold">Filter Proteins By Applying a Missing Value Threshold<a class="anchor" aria-label="anchor" href="#filter-proteins-by-applying-a-missing-value-threshold"></a>
</h4>
<p>To reduce the amount of missing values, it is possible to filter
proteins by applying a missing value threshold. The function
<code><a href="../reference/filter_NA_proteins_by_threshold.html">filter_NA_proteins_by_threshold()</a></code> removes proteins with
more missing values than the specified threshold. The threshold is a
value between 0 and 1, where 0.7, for instance, means that proteins with
less than 70% of real values will be removed, i.e., proteins with more
than 30% missing values will be removed.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_NA_proteins_by_threshold.html">filter_NA_proteins_by_threshold</a></span><span class="op">(</span><span class="va">se</span>, thr <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span> </span>
<span><span class="co">#&gt; 99 proteins were removed.</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_NA_heatmap.html">plot_NA_heatmap</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="filter-samples">Filter Samples<a class="anchor" aria-label="anchor" href="#filter-samples"></a>
</h3>
<p>Following filtering proteins by different criteria, samples can be
analyzed more in detail. PRONE provides some functions, such as
<code><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples()</a></code> and
<code><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples()</a></code>, to get an overview of the number of
proteins and the total intensity per sample, but also offers the
automatic outlier detection method of POMA.</p>
<div class="section level4">
<h4 id="quality-control">Quality Control<a class="anchor" aria-label="anchor" href="#quality-control"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_nr_prot_samples.html">plot_nr_prot_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="../reference/plot_tot_int_samples.html">plot_tot_int_samples</a></span><span class="op">(</span><span class="va">se</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="remove-samples-manually">Remove Samples Manually<a class="anchor" aria-label="anchor" href="#remove-samples-manually"></a>
</h4>
<p>Based on these plots, samples “1.HC_Pool1” and 1_HC_Pool2 seem to be
outliers. You can easily remove samples manually by using the
<code><a href="../reference/remove_samples_manually.html">remove_samples_manually()</a></code> function.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_samples_manually.html">remove_samples_manually</a></span><span class="op">(</span><span class="va">se</span>, <span class="st">"Label"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"1.HC_Pool1"</span>, <span class="st">"1.HC_Pool2"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 samples removed.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="remove-reference-samples">Remove Reference Samples<a class="anchor" aria-label="anchor" href="#remove-reference-samples"></a>
</h4>
<p>And you can remove the reference samples directly using the function
<code><a href="../reference/remove_reference_samples.html">remove_reference_samples()</a></code>. But attention: possibly you
need them for normalization! That is exactly why we currently keep
them!</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_no_refs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_reference_samples.html">remove_reference_samples</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 reference samples removed from the SummarizedExperiment object.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="outlier-detection-via-poma-r-package">Outlier Detection via POMA R Package<a class="anchor" aria-label="anchor" href="#outlier-detection-via-poma-r-package"></a>
</h4>
<p>The POMA R package provides a method to detect outliers in proteomics
data. The function <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> detects outliers
in the data based on the POMA algorithm. The function returns a list
with the following elements: polygon plot, distance boxplot, and the
outliers. For further information on the POMA algorithm, please refer to
the original publication: 10.1371/journal.pcbi.1009148.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#poma_res &lt;- detect_outliers_POMA(se, ain = "log2")</span></span>
<span></span>
<span><span class="co">#poma_res$polygon_plot</span></span>
<span></span>
<span><span class="co">#poma_res$distance_boxplot</span></span>
<span></span>
<span><span class="co">#poma_res$outliers</span></span></code></pre></div>
<p>To remove the outliers detected via the POMA algorithm, just put the
data.table of the <code><a href="../reference/detect_outliers_POMA.html">detect_outliers_POMA()</a></code> function into the
<code><a href="../reference/remove_POMA_outliers.html">remove_POMA_outliers()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#se &lt;- remove_POMA_outliers(se, poma_res$outliers)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h3>
<p>For normalization, there are multiple functions available which can
be used to normalize the data. First of all, to know which normalization
methods can be applied:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_normalization_methods.html">get_normalization_methods</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "GlobalMean"    "GlobalMedian"  "Median"        "Mean"         </span></span>
<span><span class="co">#&gt;  [5] "IRS"           "Quantile"      "VSN"           "LoessF"       </span></span>
<span><span class="co">#&gt;  [9] "LoessCyc"      "RLR"           "RlrMA"         "RlrMACyc"     </span></span>
<span><span class="co">#&gt; [13] "EigenMS"       "MAD"           "RobNorm"       "TMM"          </span></span>
<span><span class="co">#&gt; [17] "limBE"         "NormicsVSN"    "NormicsMedian"</span></span></code></pre></div>
<p>You can use <code><a href="../reference/normalize_se.html">normalize_se()</a></code> by specifying all
normalization methods you want to apply on the data. For instance, if
you want to perform median, mean, and MAD normalization, just execute
this line:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RobNorm"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, combination_pattern <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; RobNorm completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; The legacy packages maptools, rgdal, and rgeos, underpinning the sp package,</span></span>
<span><span class="co">#&gt; which was just loaded, will retire in October 2023.</span></span>
<span><span class="co">#&gt; Please refer to R-spatial evolution reports for details, especially</span></span>
<span><span class="co">#&gt; https://r-spatial.org/r/2023/05/15/evolution4.html.</span></span>
<span><span class="co">#&gt; It may be desirable to make the sf package available;</span></span>
<span><span class="co">#&gt; package maintainers should consider adding sf to Suggests:.</span></span>
<span><span class="co">#&gt; The sp package is now running under evolution status 2</span></span>
<span><span class="co">#&gt;      (status 2 uses the sf package in place of rgdal)</span></span>
<span><span class="co">#&gt; MAD completed.</span></span></code></pre></div>
<p>However, you can directly also add combinations of methods. The
combination pattern specifies which method to apply on top of the
specified normalized data. For instance, if you want to perform IRS on
top of Median-normalized data:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RobNorm"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"IRS_on_RobNorm"</span><span class="op">)</span>, combination_pattern <span class="op">=</span> <span class="st">"_on_"</span><span class="op">)</span></span>
<span><span class="co">#&gt; RobNorm completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; MAD completed.</span></span>
<span><span class="co">#&gt; IRS normalization performed on RobNorm-normalized data completed.</span></span></code></pre></div>
<p>Finally, you can also normalize your data by applying the specific
normalization method. This makes it possible to design the parameters of
an individual function more specifically. For instance, if you want to
normalize the data by the median, you can use the function
<code><a href="../reference/medianNorm.html">medianNorm()</a></code>. By default, median normalization is performed
on raw-data. Using the individual normalization functions, you can
easily specify</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/medianNorm.html">medianNorm</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="st">"log2"</span>, aout <span class="op">=</span> <span class="st">"Median"</span><span class="op">)</span></span></code></pre></div>
<p>All normalized intensities are stored in the SummarizedExperiment
object and you can check the already performed normalization techniques
using:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">se_norm</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "raw"            "log2"           "RobNorm"        "Mean"          </span></span>
<span><span class="co">#&gt; [5] "MAD"            "IRS_on_RobNorm" "Median"</span></span></code></pre></div>
<p>We suggest using the default value of the on_raw parameter. This
parameter specifies whether the data should be normalized on raw or
log2-transformed data. The default value of the “on_raw” parameters was
made for each normalization method individually based on
publications.</p>
</div>
<div class="section level3">
<h3 id="qualitative-and-quantitative-evaluation">Qualitative and Quantitative Evaluation<a class="anchor" aria-label="anchor" href="#qualitative-and-quantitative-evaluation"></a>
</h3>
<p>PRONE offers many functions to comparatively evaluate the performance
of the normalization methods. Notably, the parameter “ain” can always be
set. By specifying “ain = NULL”, all normalization methods that were
previously performed and are saved in the SummarizedExperiment object
are considered. If you want to evaluate only a selection of
normalization methods, you can specify them in the “ain” parameter. For
instance, if you want to evaluate only the IRS_on_RobNorm and Mean
normalization methods, you can set “ain = c(”IRS_on_RobNorm”,
“Mean”)“.</p>
<div class="section level4">
<h4 id="visual-inspection">Visual Inspection<a class="anchor" aria-label="anchor" href="#visual-inspection"></a>
</h4>
<p>You can comparatively visualize the normalized data by using the
function <code><a href="../reference/plot_boxplots.html">plot_boxplots()</a></code>, <code><a href="../reference/plot_densities.html">plot_densities()</a></code>,
and <code>plot_pca()</code>.</p>
<div class="section level5">
<h5 id="boxplots-of-normalized-data">Boxplots of Normalized Data<a class="anchor" aria-label="anchor" href="#boxplots-of-normalized-data"></a>
</h5>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, ncol <span class="op">=</span> <span class="fl">3</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<p>But you can also just plot a selection of normalization methods and
color for instance by batch:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"Mean"</span><span class="op">)</span>, color_by <span class="op">=</span> <span class="st">"Pool"</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
<p>Another option that you have is to return the boxplots for each
normalized data independently as a single ggplot object. For this, you
need to set facet = FALSE:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_boxplots.html">plot_boxplots</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"Mean"</span><span class="op">)</span>, color_by <span class="op">=</span> <span class="st">"Pool"</span>, label_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $IRS_on_RobNorm</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-29-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Mean</span></span></code></pre>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-29-2.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="densities-of-normalized-data">Densities of Normalized Data<a class="anchor" aria-label="anchor" href="#densities-of-normalized-data"></a>
</h5>
<p>Similarly you can visualize the densities of the normalized data.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_densities.html">plot_densities</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="pca-of-normalized-data">PCA of Normalized Data<a class="anchor" aria-label="anchor" href="#pca-of-normalized-data"></a>
</h5>
<p>Furthermore, you can visualize the normalized data in a PCA plot.
Here you have some more arguments that can be specified If you decide to
visualize the methods in independent plots (facet_norm = FALSE), then a
list of ggplot objects is returned. However, you have the additional
option to facet by any other column of the metadata (using the facet_by
parameter). Here an example:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RobNorm"</span>, <span class="st">"Mean"</span><span class="op">)</span>, color_by <span class="op">=</span> <span class="st">"Group"</span>, label_by <span class="op">=</span> <span class="st">"No"</span>, shape_by <span class="op">=</span> <span class="st">"Pool"</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span>, facet_by <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span><span class="co">#&gt; No labeling of samples.</span></span>
<span><span class="co">#&gt; $RobNorm</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-31-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $Mean</span></span></code></pre>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-31-2.png" width="700"></p>
<p>Or you can simply plot the PCA of the normalized data next to each
other. However, the facet_by argument can then not be used. Reminder, by
setting color_by = NULL, it will be first checked if a condition has
been set in the SummarizedExperiment during loading the data.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Mean"</span>, <span class="st">"log2"</span><span class="op">)</span>, color_by <span class="op">=</span> <span class="cn">NULL</span>, label_by <span class="op">=</span> <span class="st">"No"</span>, shape_by <span class="op">=</span> <span class="st">"Pool"</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; No labeling of samples.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-32-1.png" width="700"></p>
<p>Additionally, you can add all the individual normalized sample
intensities to a big SummarizedExperiment object, and perform a single
PCA on all samples (all samples meaning samples from all normalization
methods). First, you need to create the SummarizedExperiment object
using <code><a href="../reference/generate_complete_SE.html">generate_complete_SE()</a></code> and then you can simply call
the <code><a href="../reference/plot_PCA.html">plot_PCA()</a></code> function.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_complete</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generate_complete_SE.html">generate_complete_SE</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="co"># NULL -&gt; all assays are taken</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_PCA.html">plot_PCA</a></span><span class="op">(</span><span class="va">se_complete</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, color_by <span class="op">=</span> <span class="st">"Normalization"</span>, label_by <span class="op">=</span> <span class="st">"No"</span>, shape_by <span class="op">=</span> <span class="st">"Pool"</span>, facet_norm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; No labeling of samples.</span></span>
<span><span class="co">#&gt; No faceting done.</span></span>
<span><span class="co">#&gt; $all</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-33-1.png" width="700"></p>
</div>
</div>
<div class="section level4">
<h4 id="intragroup-variation">Intragroup Variation<a class="anchor" aria-label="anchor" href="#intragroup-variation"></a>
</h4>
<p>The assessment of the normalization methods is commonly centered on
their ability to decrease intragroup variation between samples, using
intragroup pooled coefficient of variation (PCV), pooled estimate of
variance (PEV), and pooled median absolute deviation (PMAD) as measures
of variability. Furthermore, the Pearson correlation between samples is
used to measure the similarity of the samples within the same group.</p>
<p>In PRONE, you can evaluate the intragroup variation of the normalized
data by using the functions <code><a href="../reference/plot_intragroup_correlation.html">plot_intragroup_correlation()</a></code>,
<code><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV()</a></code>, <code><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD()</a></code>,
and <code><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV()</a></code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_correlation.html">plot_intragroup_correlation</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, method <span class="op">=</span> <span class="st">"pearson"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-34-1.png" width="700"></p>
<p>You have two options to visualize intragroup PCV, PEV, and PMAD. You
can either simply generate boxplots of intragroup variation of each
normalization method (diff = FALSE), or you can visualize the reduction
of intragroup variation of each normalization method compared to log2
(diff = TRUE).</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-35-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-35-2.png" width="700"></p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-35-3.png" width="700"></p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PCV.html">plot_intragroup_PCV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-36-1.png" width="700"></p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PEV.html">plot_intragroup_PEV</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-36-2.png" width="700"></p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_intragroup_PMAD.html">plot_intragroup_PMAD</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, diff <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-36-3.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="subset-summarizedexperiment">Subset SummarizedExperiment<a class="anchor" aria-label="anchor" href="#subset-summarizedexperiment"></a>
</h4>
<p>After the qualitative and quantitative evaluation, you may have
noticed that some normalization techniques are not appropriate for the
specific real-world data set. For further analysis, you want to remove
them and not evaluate the specific normalization methods furthermore.
For this, the <code><a href="../reference/remove_assays_from_SE.html">remove_assays_from_SE()</a></code> method can be
used.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_no_MAD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_assays_from_SE.html">remove_assays_from_SE</a></span><span class="op">(</span><span class="va">se_norm</span>, assays_to_remove <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"MAD"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>In contrast, you can also subset the SummarizedExperiment object to
only include specific normalization techniques using the
<code><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm()</a></code>method.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_subset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/subset_SE_by_norm.html">subset_SE_by_norm</a></span><span class="op">(</span><span class="va">se_norm</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Mean"</span>, <span class="st">"RobNorm"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="imputation">Imputation<a class="anchor" aria-label="anchor" href="#imputation"></a>
</h3>
<p>TODO</p>
</div>
<div class="section level3">
<h3 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h3>
<p>After having performed normalization and evaluated the different
normalization methods via qualitative and quantitative analysis,
differential expression analysis can be used to further analyze the
differences of the normalization methods.</p>
<p>However before, you need to remove the reference samples in case of a
TMT experiment. This can be easily done with the function
<code><a href="../reference/remove_reference_samples.html">remove_reference_samples()</a></code>.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/remove_reference_samples.html">remove_reference_samples</a></span><span class="op">(</span><span class="va">se_norm</span><span class="op">)</span></span>
<span><span class="co">#&gt; 2 reference samples removed from the SummarizedExperiment object.</span></span></code></pre></div>
<div class="section level4">
<h4 id="run-de-analysis">Run DE Analysis<a class="anchor" aria-label="anchor" href="#run-de-analysis"></a>
</h4>
<p>First, you need to specify the comparisons you want to perform in DE
analysis. For this, the function <code>specifc_comparisons()</code> was
developed which helps to build the right comparison strings. For
instance, if you are having a condition column “Condition” in your meta
data that is built upon two individual groups, such as “diabetic_3d”
referring to diabetic samples at day 3 after operation, you can simply
specify sep = “_” and the method only considers comparisons where at
least one of the group remains static.</p>
<p>However, you can also just simply create a vector of comparisons to
ensure the correct order and handle this vector over to the DE analysis
method.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/specify_comparisons.html">specify_comparisons</a></span><span class="op">(</span><span class="va">se_norm</span>, condition <span class="op">=</span> <span class="st">"Group"</span>, sep <span class="op">=</span> <span class="st">"_"</span>, control <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>The function <code><a href="../reference/run_DE.html">run_DE()</a></code> performs the DE analysis on the
selected SummarizedExperiment and comparisons. DE analysis can be
performed on multiple assays (normalization methods) at once using the
already known “ain” parameter. The condition of the SummarizedExperiment
object, specified at the beginning, can be used (condition = NULL) or
any other column of the meta data can specified. Three methods are
available for DE analysis: limma, DEqMS, and ROTS. The meaning of the
other parameters can be extracted from the documentation of the
method.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DE.html">run_DE</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">se_norm</span>, </span>
<span>                     comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>                     ain <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     DE_method <span class="op">=</span> <span class="st">"limma"</span>, </span>
<span>                     covariate <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     logFC <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     logFC_up <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     logFC_down <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                     p_adj <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                     B <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                     K <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; DE Analysis will not be performed on raw data.</span></span>
<span><span class="co">#&gt; log2: DE analysis completed.</span></span>
<span><span class="co">#&gt; RobNorm: DE analysis completed.</span></span>
<span><span class="co">#&gt; Mean: DE analysis completed.</span></span>
<span><span class="co">#&gt; MAD: DE analysis completed.</span></span>
<span><span class="co">#&gt; IRS_on_RobNorm: DE analysis completed.</span></span>
<span><span class="co">#&gt; Median: DE analysis completed.</span></span></code></pre></div>
<p>If you want to apply other logFC or p-value threshold, there is no
need to re-run the DE analysis again. With
<code><a href="../reference/apply_thresholds.html">apply_thresholds()</a></code>, you can simply change the threshold
values.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">new_de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/apply_thresholds.html">apply_thresholds</a></span><span class="op">(</span>de_res <span class="op">=</span> <span class="va">de_res</span>, logFC <span class="op">=</span> <span class="cn">FALSE</span>, p_adj <span class="op">=</span> <span class="cn">TRUE</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="visualize-de-results">Visualize DE Results<a class="anchor" aria-label="anchor" href="#visualize-de-results"></a>
</h4>
<div class="section level5">
<h5 id="barplot">Barplot<a class="anchor" aria-label="anchor" href="#barplot"></a>
</h5>
<p>To get an overview of the DE results of the different normalization
methods, you can visualize the number of significant DE proteins per
normalization method in a barplot using
<code><a href="../reference/plot_overview_DE_bar.html">plot_overview_DE_bar()</a></code>. This plot can be generated in
different ways by specifying the “plot_type” parameter: * single: plot a
single barplot for each comparison (default) * facet_comp: facet the
barplot by comparison * stacked: stack the number of DE per comparison *
facet_regulation: stack the number of DE per comparison but facet by up-
and down-regulated</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_overview_DE_bar.html">plot_overview_DE_bar</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, plot_type <span class="op">=</span> <span class="st">"facet_regulation"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-43-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="tile-plot">Tile Plot<a class="anchor" aria-label="anchor" href="#tile-plot"></a>
</h5>
<p>You can also get an overview of the DE results in form of a heatmap
using the <code><a href="../reference/plot_overview_DE_tile.html">plot_overview_DE_tile()</a></code>.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_overview_DE_tile.html">plot_overview_DE_tile</a></span><span class="op">(</span><span class="va">de_res</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-44-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="volcano-plots">Volcano Plots<a class="anchor" aria-label="anchor" href="#volcano-plots"></a>
</h5>
<p>Another option is to generate volcano plots for each comparison. The
function <code><a href="../reference/plot_volcano_DE.html">plot_volcano_DE()</a></code> generates a grid of volcano
plots for each normalization techniques (facet_norm = TRUE) or for each
comparison (facet_comparison = TRUE). A list of volcano plots is
returned.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_volcano_DE.html">plot_volcano_DE</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>, facet_norm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span>
<span><span class="co">#&gt; $`Healthy.control-Pulmonary.tuberculosis`</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-45-1.png" width="700"></p>
</div>
<div class="section level5">
<h5 id="coverage-of-biomarkers">Coverage of Biomarkers<a class="anchor" aria-label="anchor" href="#coverage-of-biomarkers"></a>
</h5>
<p>In case you know some biomarkers of previous publications that should
be significantly expressed in your biological context, you can check the
number of specified markers that are DE in the different normalization
methods.</p>
<p>TODO: plot_coverage_DE_bar</p>
</div>
</div>
<div class="section level4">
<h4 id="intersection-analysis-of-de-results">Intersection Analysis of DE Results<a class="anchor" aria-label="anchor" href="#intersection-analysis-of-de-results"></a>
</h4>
<p>Moreover, you can also intersect the DE results of different
normalization methods to see how many DE proteins overlap. You can
either plot for each requested comparison an individual upset plot
(plot_type = “single”) or stack the number of overlapping DE proteins
per comparison (“stacked”). Not only the upset plot(s) are returned, but
also a table with the intersections is provided by the
<code><a href="../reference/plot_upset_DE.html">plot_upset_DE()</a></code>function.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intersections</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_upset_DE.html">plot_upset_DE</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, min_degree <span class="op">=</span> <span class="fl">2</span>, plot_type <span class="op">=</span> <span class="st">"stacked"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span>
<span><span class="va">intersections</span><span class="op">$</span><span class="va">upset</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-46-1.png" width="700"></p>
<p>Additionally, the Jaccard similarity index can be calculated to
quantify the similarity of the DE results between the different
normalization methods. A individual heatmap can be generated for each
comparison (“plot_type = single”), a single heatmap facetted by
comparison (“facet_comp”) or a single heatmap taking all comparisons
into account (“all”) can be generated.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_jaccard_heatmap.html">plot_jaccard_heatmap</a></span><span class="op">(</span><span class="va">de_res</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="va">comparisons</span>, plot_type <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-47-1.png" width="700"></p>
</div>
</div>
<div class="section level3">
<h3 id="download-data">Download Data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h3>
<p>Finally, you can easily download the normalized data by using the
<code><a href="../reference/export_data.html">export_data()</a></code>function. This function will save the
specified assays as CSV files and the SummarizedExperiment object as an
RDS file in a specified output directory.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/export_data.html">export_data</a></span><span class="op">(</span><span class="va">se_norm</span>, out_dir <span class="op">=</span> <span class="st">"output/"</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"IRS_on_RobNorm"</span>, <span class="st">"Mean"</span>, <span class="st">"RobNorm"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="spike-in-data-sets">Spike-in Data Sets<a class="anchor" aria-label="anchor" href="#spike-in-data-sets"></a>
</h2>
<p>PRONE also offers some additional functionalities for the evaluation
of normalization techniques on spike-in data sets with a known ground
truth. These functionalities will be delineated subsequently.
Additionally, all functionalities detailed in the context of real-world
data sets remain applicable to the SummarizedExperiment object
associated with spike-in data sets.</p>
<p>The example spike-in data set is from:</p>
<p>Jürgen Cox, Marco Y. Hein, Christian A. Luber, Igor Paron, Nagarjuna
Nagaraj, and Matthias Mann. Accurate Proteome-wide Label-free
Quantification by Delayed Normalization and Maximal Peptide Ratio
Extraction, Termed MaxLFQ. Molecular &amp; Cellular Proteomics 13.9
(Sept. 2014), pp. 2513–2526. <a href="https://doi.org/10.1074/mcp.M113.031591" class="external-link uri">https://doi.org/10.1074/mcp.M113.031591</a>.</p>
<div class="section level3">
<h3 id="load-data-1">Load Data<a class="anchor" aria-label="anchor" href="#load-data-1"></a>
</h3>
<p>Initially, the spike-in data set needs to be imported into a
SummarizedExperiment container. In contrast to the real-world data sets,
you need to specify the “spike_column”, “spike_value”,
“spike_concentration”, and utilize the <code><a href="../reference/load_spike_data.html">load_spike_data()</a></code>
function for this purpose.</p>
<p>Here, the “spike_column” denotes the column in the protein intensity
data file encompassing information whether a proteins is classified as
spike-in or background protein. The “spike_value” determines the
value/identifier that is used to identify the spike-in proteins in the
“spike_column”, and the “spike_concentration” identifies the column
containing the concentration values of the spike-in proteins (in this
case the different conditions that will be tested in DE analysis).</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../inst/extdata/spike/proteinGroups.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="st">"../inst/extdata/spike/experimentalDesign.txt"</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if some protein groups are mixed</span></span>
<span><span class="va">mixed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Homo sapiens.*Escherichia|Escherichia.*Homo sapiens"</span>, <span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="op">!</span><span class="va">mixed</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"HUMAN"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"ECOLI"</span>, <span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"ECOLI"</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_spike_data.html">load_spike_data</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">md</span>, spike_column <span class="op">=</span> <span class="st">"Spiked"</span>, spike_value <span class="op">=</span> <span class="st">"ECOLI"</span>, spike_concentration <span class="op">=</span> <span class="st">"Concentration"</span>,protein_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, gene_column <span class="op">=</span> <span class="st">"Gene.names"</span>, ref_samples <span class="op">=</span> <span class="cn">NULL</span>, batch_column <span class="op">=</span> <span class="cn">NULL</span>, condition_column <span class="op">=</span> <span class="st">"Condition"</span>, label_column <span class="op">=</span> <span class="st">"Label"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="overview-of-the-data-1">Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data-1"></a>
</h3>
<p>To get an overview on the number of identified (non-NA) spike-in
proteins per sample, you can use the
<code><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins()</a></code> function.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-50-1.png" width="700">
To compare the distributions of spike-in and human proteins in the
different sample groups (here high-low), use the function
<code><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked()</a></code>. Again, “condition = NULL” means
that the condition specified by loading the data is used, but you can
also specify any other column of the meta data.</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-51-1.png" width="700"></p>
<p>If you want to have a look at the amount of actual measure spike-in,
you can use the <code><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked()</a></code> function. Moreover,
you can analyze whether the intensities of the background proteins, here
HUMAN proteins, are constant across the different spike-in
concentrations.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked</a></span><span class="op">(</span><span class="va">se</span>, xlab <span class="op">=</span> <span class="st">"Concentration"</span><span class="op">)</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-52-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h3>
<p>Given that the preprocessing (including filtering of proteins and
samples), normalization, and imputation operations remain invariant for
spike-in data sets compared to real-world data sets, the same
methodologies can be employed across both types. In this context,
normalization will only be demonstrated here as the performance of the
methods will be evaluated in DE analysis, while detailed descriptions of
the other functionalities are available in preceding sections.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, combination_pattern <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; MAD completed.</span></span>
<span><span class="co">#&gt; LoessF completed.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-expression-analysis-1">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis-1"></a>
</h3>
<p>Due to the known spike-in concentrations, the normalization methods
can be evaluated based on their ability to detect DE proteins. The DE
analysis can be conducted using the same methodology as for real-world
data sets. However, other visualization options are available and
performance metrics can be calculated for spike-in data sets.</p>
<div class="section level4">
<h4 id="run-de-analysis-1">Run DE Analysis<a class="anchor" aria-label="anchor" href="#run-de-analysis-1"></a>
</h4>
<p>First, you need to specify the comparisons you want to perform in DE
analysis. For this, a special function was developed which helps to
build the right comparison strings.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/specify_comparisons.html">specify_comparisons</a></span><span class="op">(</span><span class="va">se_norm</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, sep <span class="op">=</span> <span class="cn">NULL</span>, control <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>Then you can run DE analysis:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DE.html">run_DE</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">se_norm</span>, </span>
<span>                     comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>                     ain <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     DE_method <span class="op">=</span> <span class="st">"limma"</span>, </span>
<span>                     covariate <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     logFC <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     logFC_up <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     logFC_down <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                     p_adj <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                     B <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                     K <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; DE Analysis will not be performed on raw data.</span></span>
<span><span class="co">#&gt; log2: DE analysis completed.</span></span>
<span><span class="co">#&gt; Median: DE analysis completed.</span></span>
<span><span class="co">#&gt; Mean: DE analysis completed.</span></span>
<span><span class="co">#&gt; MAD: DE analysis completed.</span></span>
<span><span class="co">#&gt; LoessF: DE analysis completed.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="evaluate-de-results-with-performance-metrics">Evaluate DE Results with Performance Metrics<a class="anchor" aria-label="anchor" href="#evaluate-de-results-with-performance-metrics"></a>
</h4>
<p>Before being able to visualize the DE results, you need to run
<code><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE()</a></code> to calculate multiple performance
metrics, such as the number of true positives, number fo false
positives, etc.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span><span class="op">)</span></span></code></pre></div>
<p>You have different options to visualize the amount of true and false
positives for the different normalization techniques and pairwise
comparisons. For all these functions, you can specify a subset of
normalization methods and comparisons. If you do not specify anything,
all normalization methods and comparisons of the “stats” data frame are
used.</p>
<p>The <code><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar()</a></code> function generates a barplot
showing the number of false positives and true positives for each
normalization method and is facetted by pairwise comparison.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-57-1.png" width="700"></p>
<p>If many pairwise comparisons were performed, the
<code><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box()</a></code> function can be used to visualize
the distribution of true and false positives for all pairwise
comparisons in a boxplot.</p>
<p>Given that the data set encompasses merely two distinct spike-in
concentrations and therefore only one pairwise comparison has been
conducted in DE analysis, a barplot would be more appropriate.
Nonetheless, for demonstration purposes, a boxplot will be used
here.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-58-1.png" width="700"></p>
<p>Furthermore, similarly the <code><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter()</a></code>
function can be used to visualize the true and false positives in a
scatter plot. Here, a scatterplot of the median true positives and false
positives is calculated across all comparisons and displayed for each
normalization method with errorbars showing the Q1 and Q3. Here again,
this plot is more suitable for data sets with multiple pairwise
comparisons.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-59-1.png" width="700"></p>
<p>Furthermore, other performance metrics that the number of true
positives and false positives can be visualized using the
<code><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap()</a></code> function. Here, currently, the
sensitivity, specificity, precision, FPR, F1 score, and accuracy can be
visualized for the different normalization methods and pairwise
comparisons.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Precision"</span>, <span class="st">"F1Score"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-60-1.png" width="700"></p>
<p>Finally, ROC and AUC values can be calculated and visualized using
the <code><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked()</a></code> function. This function returns a
plot showing the ROC curves, a bar plot with the AUC values for each
comparison, and a boxplot with the distributions of AUC values across
all comparisons.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; $ROC</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-61-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_bars</span></span></code></pre>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-61-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_box</span></span></code></pre>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-61-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_dt</span></span>
<span><span class="co">#&gt;   PANEL Comparison group  Assay       AUC</span></span>
<span><span class="co">#&gt; 1     1        H-L     1 LoessF 0.8262511</span></span>
<span><span class="co">#&gt; 2     1        H-L     2   Mean 0.6221822</span></span>
<span><span class="co">#&gt; 3     1        H-L     3 Median 0.8026922</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="log-fold-change-distributions">Log Fold Change Distributions<a class="anchor" aria-label="anchor" href="#log-fold-change-distributions"></a>
</h4>
<p>Furthermore, you can visualize the distribution of log fold changes
for the different conditions using the
<code><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked()</a></code> function. The fold changes of
the background proteins should be centered around zero, while the
spike-in proteins should be centered around the actual log fold change
calculated based on the spike-in concentrations.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-62-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="p-value-distributions">P-Value Distributions<a class="anchor" aria-label="anchor" href="#p-value-distributions"></a>
</h4>
<p>Similarly, the distributions of the p-values can be visualized using
the <code><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked()</a></code> function.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-63-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="log-fold-change-thresholds">Log Fold Change Thresholds<a class="anchor" aria-label="anchor" href="#log-fold-change-thresholds"></a>
</h4>
<p>Due to the high amount of false positives encountered in spike-in
data sets, we provided a function to test for different log fold change
thresholds to try to reduce the amount of false positives. The function
<code><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked()</a></code> can be used to visualize the
number of true positives and false positives for different log fold
change thresholds.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $TP</span></span></code></pre></div>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-64-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FP</span></span></code></pre>
<p><img src="PRONE.R_files/figure-html/unnamed-chunk-64-2.png" width="700"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
