<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="PRONE">
<title>How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data • PRONE</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data">
<meta property="og:description" content="PRONE">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">PRONE</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.30</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/PRONE.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Preprocessing.html">Preprocessing</a>
    <a class="dropdown-item" href="../articles/Normalization.html">Normalization</a>
    <a class="dropdown-item" href="../articles/Imputation.html">Imputation</a>
    <a class="dropdown-item" href="../articles/Differential_Expression.html">Differential Expression Analysis</a>
    <a class="dropdown-item" href="../articles/Spike_In_Data.html">How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/lisiarend/PRONE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>How to Use PRONE for the Evaluation of Normalization Methods on Spike-In Data</h1>
                        <h4 data-toc-skip class="author">Arend Lis</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/lisiarend/PRONE/blob/HEAD/vignettes/Spike_In_Data.Rmd" class="external-link"><code>vignettes/Spike_In_Data.Rmd</code></a></small>
      <div class="d-none name"><code>Spike_In_Data.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lisiarend/PRONE" class="external-link">PRONE</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="spike-in-data-sets">Spike-in Data Sets<a class="anchor" aria-label="anchor" href="#spike-in-data-sets"></a>
</h2>
<p>PRONE also offers some additional functionalities for the evaluation
of normalization techniques on spike-in data sets with a known ground
truth. These functionalities will be delineated subsequently.
Additionally, all functionalities detailed in the context of real-world
data sets remain applicable to the SummarizedExperiment object
associated with spike-in data sets.</p>
<p>The example spike-in data set is from <span class="citation">(Cox et
al. 2014)</span>.</p>
<div class="section level3">
<h3 id="load-data">Load Data<a class="anchor" aria-label="anchor" href="#load-data"></a>
</h3>
<p>Initially, the spike-in data set needs to be imported into a
SummarizedExperiment container. In contrast to the real-world data sets,
you need to specify the “spike_column”, “spike_value”,
“spike_concentration”, and utilize the <code><a href="../reference/load_spike_data.html">load_spike_data()</a></code>
function for this purpose.</p>
<p>Here, the “spike_column” denotes the column in the protein intensity
data file encompassing information whether a proteins is classified as
spike-in or background protein. The “spike_value” determines the
value/identifier that is used to identify the spike-in proteins in the
“spike_column”, and the “spike_concentration” identifies the column
containing the concentration values of the spike-in proteins (in this
case the different conditions that will be tested in DE analysis).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"spike_in_proteinGroups.txt"</span><span class="op">)</span></span>
<span><span class="va">md_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readPRONE_example.html">readPRONE_example</a></span><span class="op">(</span><span class="st">"spike_in_metadata.txt"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">data_path</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span><span class="va">md</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="va">md_path</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check if some protein groups are mixed</span></span>
<span><span class="va">mixed</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"Homo sapiens.*Escherichia|Escherichia.*Homo sapiens"</span>, <span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">[</span><span class="op">!</span><span class="va">mixed</span>,<span class="op">]</span></span>
<span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"HUMAN"</span>, <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">data</span><span class="op">$</span><span class="va">Spiked</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"ECOLI"</span>, <span class="va">data</span><span class="op">$</span><span class="va">Fasta.headers</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"ECOLI"</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_spike_data.html">load_spike_data</a></span><span class="op">(</span><span class="va">data</span>, <span class="va">md</span>, spike_column <span class="op">=</span> <span class="st">"Spiked"</span>, spike_value <span class="op">=</span> <span class="st">"ECOLI"</span>, spike_concentration <span class="op">=</span> <span class="st">"Concentration"</span>,protein_column <span class="op">=</span> <span class="st">"Protein.IDs"</span>, gene_column <span class="op">=</span> <span class="st">"Gene.names"</span>, ref_samples <span class="op">=</span> <span class="cn">NULL</span>, batch_column <span class="op">=</span> <span class="cn">NULL</span>, condition_column <span class="op">=</span> <span class="st">"Condition"</span>, label_column <span class="op">=</span> <span class="st">"Label"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="overview-of-the-data">Overview of the Data<a class="anchor" aria-label="anchor" href="#overview-of-the-data"></a>
</h3>
<p>To get an overview on the number of identified (non-NA) spike-in
proteins per sample, you can use the
<code><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins()</a></code> function.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_identified_spiked_proteins.html">plot_identified_spiked_proteins</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; Label of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-3-1.png" width="700">
To compare the distributions of spike-in and human proteins in the
different sample groups (here high-low), use the function
<code><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked()</a></code>. Again, “condition = NULL” means
that the condition specified by loading the data is used, but you can
also specify any other column of the meta data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_histogram_spiked.html">plot_histogram_spiked</a></span><span class="op">(</span><span class="va">se</span>, condition <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>If you want to have a look at the amount of actual measure spike-in,
you can use the <code><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked()</a></code> function. Moreover,
you can analyze whether the intensities of the background proteins, here
HUMAN proteins, are constant across the different spike-in
concentrations.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_profiles_spiked.html">plot_profiles_spiked</a></span><span class="op">(</span><span class="va">se</span>, xlab <span class="op">=</span> <span class="st">"Concentration"</span><span class="op">)</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="preprocessing-normalization-imputation">Preprocessing, Normalization, &amp; Imputation<a class="anchor" aria-label="anchor" href="#preprocessing-normalization-imputation"></a>
</h3>
<p>Given that the preprocessing (including filtering of proteins and
samples), normalization, and imputation operations remain invariant for
spike-in data sets compared to real-world data sets, the same
methodologies can be employed across both types. In this context,
normalization will only be demonstrated here as the performance of the
methods will be evaluated in DE analysis, while detailed descriptions of
the other functionalities are available in preceding sections.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_norm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize_se.html">normalize_se</a></span><span class="op">(</span><span class="va">se</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, combination_pattern <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; Median completed.</span></span>
<span><span class="co">#&gt; Mean completed.</span></span>
<span><span class="co">#&gt; MAD completed.</span></span>
<span><span class="co">#&gt; LoessF completed.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="differential-expression-analysis">Differential Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-expression-analysis"></a>
</h3>
<p>Due to the known spike-in concentrations, the normalization methods
can be evaluated based on their ability to detect DE proteins. The DE
analysis can be conducted using the same methodology as for real-world
data sets. However, other visualization options are available and
performance metrics can be calculated for spike-in data sets.</p>
<div class="section level4">
<h4 id="run-de-analysis">Run DE Analysis<a class="anchor" aria-label="anchor" href="#run-de-analysis"></a>
</h4>
<p>First, you need to specify the comparisons you want to perform in DE
analysis. For this, a special function was developed which helps to
build the right comparison strings.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comparisons</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/specify_comparisons.html">specify_comparisons</a></span><span class="op">(</span><span class="va">se_norm</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, sep <span class="op">=</span> <span class="cn">NULL</span>, control <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<p>Then you can run DE analysis:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">de_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_DE.html">run_DE</a></span><span class="op">(</span>se <span class="op">=</span> <span class="va">se_norm</span>, </span>
<span>                     comparisons <span class="op">=</span> <span class="va">comparisons</span>,</span>
<span>                     ain <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     condition <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     DE_method <span class="op">=</span> <span class="st">"limma"</span>, </span>
<span>                     covariate <span class="op">=</span> <span class="cn">NULL</span>, </span>
<span>                     logFC <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     logFC_up <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     logFC_down <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, </span>
<span>                     p_adj <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     alpha <span class="op">=</span> <span class="fl">0.05</span>, </span>
<span>                     B <span class="op">=</span> <span class="fl">100</span>, </span>
<span>                     K <span class="op">=</span> <span class="fl">500</span><span class="op">)</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; All assays of the SummarizedExperiment will be used.</span></span>
<span><span class="co">#&gt; DE Analysis will not be performed on raw data.</span></span>
<span><span class="co">#&gt; log2: DE analysis completed.</span></span>
<span><span class="co">#&gt; Median: DE analysis completed.</span></span>
<span><span class="co">#&gt; Mean: DE analysis completed.</span></span>
<span><span class="co">#&gt; MAD: DE analysis completed.</span></span>
<span><span class="co">#&gt; LoessF: DE analysis completed.</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="evaluate-de-results-with-performance-metrics">Evaluate DE Results with Performance Metrics<a class="anchor" aria-label="anchor" href="#evaluate-de-results-with-performance-metrics"></a>
</h4>
<p>Before being able to visualize the DE results, you need to run
<code><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE()</a></code> to calculate multiple performance
metrics, such as the number of true positives, number fo false
positives, etc.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_spiked_stats_DE.html">get_spiked_stats_DE</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span><span class="op">)</span></span></code></pre></div>
<p>You have different options to visualize the amount of true and false
positives for the different normalization techniques and pairwise
comparisons. For all these functions, you can specify a subset of
normalization methods and comparisons. If you do not specify anything,
all normalization methods and comparisons of the “stats” data frame are
used.</p>
<p>The <code><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar()</a></code> function generates a barplot
showing the number of false positives and true positives for each
normalization method and is facetted by pairwise comparison.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_bar.html">plot_TP_FP_spiked_bar</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>If many pairwise comparisons were performed, the
<code><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box()</a></code> function can be used to visualize
the distribution of true and false positives for all pairwise
comparisons in a boxplot.</p>
<p>Given that the data set encompasses merely two distinct spike-in
concentrations and therefore only one pairwise comparison has been
conducted in DE analysis, a barplot would be more appropriate.
Nonetheless, for demonstration purposes, a boxplot will be used
here.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_box.html">plot_TP_FP_spiked_box</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>Furthermore, similarly the <code><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter()</a></code>
function can be used to visualize the true and false positives in a
scatter plot. Here, a scatterplot of the median true positives and false
positives is calculated across all comparisons and displayed for each
normalization method with errorbars showing the Q1 and Q3. Here again,
this plot is more suitable for data sets with multiple pairwise
comparisons.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_TP_FP_spiked_scatter.html">plot_TP_FP_spiked_scatter</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="cn">NULL</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span>
<span><span class="co">#&gt; All normalization methods of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>Furthermore, other performance metrics that the number of true
positives and false positives can be visualized using the
<code><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap()</a></code> function. Here, currently, the
sensitivity, specificity, precision, FPR, F1 score, and accuracy can be
visualized for the different normalization methods and pairwise
comparisons.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stats_spiked_heatmap.html">plot_stats_spiked_heatmap</a></span><span class="op">(</span><span class="va">stats</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, metrics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Precision"</span>, <span class="st">"F1Score"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of stats will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>Finally, ROC and AUC values can be calculated and visualized using
the <code><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked()</a></code> function. This function returns a
plot showing the ROC curves, a bar plot with the AUC values for each
comparison, and a boxplot with the distributions of AUC values across
all comparisons.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_ROC_AUC_spiked.html">plot_ROC_AUC_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"LoessF"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; $ROC</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_bars</span></span></code></pre>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-14-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_box</span></span></code></pre>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-14-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $AUC_dt</span></span>
<span><span class="co">#&gt;   PANEL Comparison group  Assay       AUC</span></span>
<span><span class="co">#&gt; 1     1        H-L     1 LoessF 0.8262511</span></span>
<span><span class="co">#&gt; 2     1        H-L     2   Mean 0.6221822</span></span>
<span><span class="co">#&gt; 3     1        H-L     3 Median 0.8026922</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="log-fold-change-distributions">Log Fold Change Distributions<a class="anchor" aria-label="anchor" href="#log-fold-change-distributions"></a>
</h4>
<p>Furthermore, you can visualize the distribution of log fold changes
for the different conditions using the
<code><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked()</a></code> function. The fold changes of
the background proteins should be centered around zero, while the
spike-in proteins should be centered around the actual log fold change
calculated based on the spike-in concentrations.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_fold_changes_spiked.html">plot_fold_changes_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="st">"Condition"</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="p-value-distributions">P-Value Distributions<a class="anchor" aria-label="anchor" href="#p-value-distributions"></a>
</h4>
<p>Similarly, the distributions of the p-values can be visualized using
the <code><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked()</a></code> function.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_pvalues_spiked.html">plot_pvalues_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, comparisons <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="log-fold-change-thresholds">Log Fold Change Thresholds<a class="anchor" aria-label="anchor" href="#log-fold-change-thresholds"></a>
</h4>
<p>Due to the high amount of false positives encountered in spike-in
data sets, we provided a function to test for different log fold change
thresholds to try to reduce the amount of false positives. The function
<code><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked()</a></code> can be used to visualize the
number of true positives and false positives for different log fold
change thresholds.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_logFC_thresholds_spiked.html">plot_logFC_thresholds_spiked</a></span><span class="op">(</span><span class="va">se_norm</span>, <span class="va">de_res</span>, condition <span class="op">=</span> <span class="cn">NULL</span>, ain <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Median"</span>, <span class="st">"Mean"</span>, <span class="st">"MAD"</span><span class="op">)</span>, nrow <span class="op">=</span> <span class="fl">1</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; All comparisons of de_res will be visualized.</span></span>
<span><span class="co">#&gt; Condition of SummarizedExperiment used!</span></span>
<span><span class="co">#&gt; $TP</span></span></code></pre></div>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $FP</span></span></code></pre>
<p><img src="Spike_In_Data_files/figure-html/unnamed-chunk-17-2.png" width="700"></p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices datasets  utils     methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] PRONE_0.1.30</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] rlang_1.1.3                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [3] clue_0.3-65                 matrixStats_1.3.0          </span></span>
<span><span class="co">#&gt;   [5] compiler_4.4.0              systemfonts_1.0.6          </span></span>
<span><span class="co">#&gt;   [7] vctrs_0.6.5                 ProtGenerics_1.35.4        </span></span>
<span><span class="co">#&gt;   [9] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [11] fastmap_1.1.1               XVector_0.43.1             </span></span>
<span><span class="co">#&gt;  [13] labeling_0.4.3              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.26              UCSC.utils_0.99.7          </span></span>
<span><span class="co">#&gt;  [17] preprocessCore_1.65.0       ragg_1.3.0                 </span></span>
<span><span class="co">#&gt;  [19] purrr_1.0.2                 xfun_0.43                  </span></span>
<span><span class="co">#&gt;  [21] MultiAssayExperiment_1.29.1 zlibbioc_1.49.3            </span></span>
<span><span class="co">#&gt;  [23] cachem_1.0.8                GenomeInfoDb_1.39.14       </span></span>
<span><span class="co">#&gt;  [25] jsonlite_1.8.8              highr_0.10                 </span></span>
<span><span class="co">#&gt;  [27] DelayedArray_0.29.9         BiocParallel_1.37.1        </span></span>
<span><span class="co">#&gt;  [29] parallel_4.4.0              cluster_2.1.6              </span></span>
<span><span class="co">#&gt;  [31] R6_2.5.1                    RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [33] bslib_0.7.0                 limma_3.59.10              </span></span>
<span><span class="co">#&gt;  [35] GenomicRanges_1.55.4        jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [37] iterators_1.0.14            Rcpp_1.0.12                </span></span>
<span><span class="co">#&gt;  [39] SummarizedExperiment_1.33.3 knitr_1.46                 </span></span>
<span><span class="co">#&gt;  [41] IRanges_2.37.1              splines_4.4.0              </span></span>
<span><span class="co">#&gt;  [43] Matrix_1.7-0                igraph_2.0.3               </span></span>
<span><span class="co">#&gt;  [45] tidyselect_1.2.1            rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [47] abind_1.4-5                 yaml_2.3.8                 </span></span>
<span><span class="co">#&gt;  [49] ggtext_0.1.2                doParallel_1.0.17          </span></span>
<span><span class="co">#&gt;  [51] codetools_0.2-20            affy_1.81.0                </span></span>
<span><span class="co">#&gt;  [53] lattice_0.22-6              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [55] plyr_1.8.9                  withr_3.0.0                </span></span>
<span><span class="co">#&gt;  [57] Biobase_2.63.1              evaluate_0.23              </span></span>
<span><span class="co">#&gt;  [59] desc_1.4.3                  xml2_1.3.6                 </span></span>
<span><span class="co">#&gt;  [61] pillar_1.9.0                affyio_1.73.0              </span></span>
<span><span class="co">#&gt;  [63] BiocManager_1.30.22         MatrixGenerics_1.15.1      </span></span>
<span><span class="co">#&gt;  [65] renv_1.0.7                  foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [67] stats4_4.4.0                MSnbase_2.29.4             </span></span>
<span><span class="co">#&gt;  [69] MALDIquant_1.22.2           ncdf4_1.22                 </span></span>
<span><span class="co">#&gt;  [71] generics_0.1.3              S4Vectors_0.41.7           </span></span>
<span><span class="co">#&gt;  [73] ggplot2_3.5.1               munsell_0.5.1              </span></span>
<span><span class="co">#&gt;  [75] scales_1.3.0                glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [77] lazyeval_0.2.2              tools_4.4.0                </span></span>
<span><span class="co">#&gt;  [79] data.table_1.15.4           mzID_1.41.0                </span></span>
<span><span class="co">#&gt;  [81] QFeatures_1.13.7            vsn_3.71.1                 </span></span>
<span><span class="co">#&gt;  [83] mzR_2.37.3                  fs_1.6.4                   </span></span>
<span><span class="co">#&gt;  [85] XML_3.99-0.16.1             grid_4.4.0                 </span></span>
<span><span class="co">#&gt;  [87] impute_1.77.0               tidyr_1.3.1                </span></span>
<span><span class="co">#&gt;  [89] MsCoreUtils_1.15.7          colorspace_2.1-0           </span></span>
<span><span class="co">#&gt;  [91] GenomeInfoDbData_1.2.12     PSMatch_1.7.2              </span></span>
<span><span class="co">#&gt;  [93] cli_3.6.2                   textshaping_0.3.7          </span></span>
<span><span class="co">#&gt;  [95] plotROC_2.3.1               fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [97] S4Arrays_1.3.7              dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [99] AnnotationFilter_1.27.0     pcaMethods_1.95.0          </span></span>
<span><span class="co">#&gt; [101] gtable_0.3.5                sass_0.4.9                 </span></span>
<span><span class="co">#&gt; [103] digest_0.6.35               BiocGenerics_0.49.1        </span></span>
<span><span class="co">#&gt; [105] SparseArray_1.3.5           farver_2.1.1               </span></span>
<span><span class="co">#&gt; [107] htmlwidgets_1.6.4           memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [109] htmltools_0.5.8.1           pkgdown_2.0.9              </span></span>
<span><span class="co">#&gt; [111] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [113] NormalyzerDE_1.21.0         statmod_1.5.0              </span></span>
<span><span class="co">#&gt; [115] gridtext_0.1.5              MASS_7.3-60.2</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-cox_accurate_2014" class="csl-entry">
Cox, Jürgen, Marco Y. Hein, Christian A. Luber, Igor Paron, Nagarjuna
Nagaraj, and Matthias Mann. 2014. <span>“Accurate Proteome-Wide
Label-Free Quantification by Delayed Normalization and Maximal Peptide
Ratio Extraction, Termed Maxlfq.”</span> <em>Molecular &amp; Cellular
Proteomics</em> 13 (9): 2513–26. <a href="https://doi.org/10.1074/mcp.M113.031591" class="external-link">https://doi.org/10.1074/mcp.M113.031591</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lis Arend.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
